<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="heinin" />

<meta name="date" content="2025-02-04" />

<title>Cell type annotations for PIPAC data</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIPAC_spatial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Cell type annotations for PIPAC data</h1>
<h4 class="author">heinin</h4>
<h4 class="date">2025-02-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>PIPAC_spatial/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240917code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240917)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240917code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240917)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/
</td>
<td style="text-align:left;">
.
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/code/PIPAC_colors_themes.R
</td>
<td style="text-align:left;">
code/PIPAC_colors_themes.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/code/plot_functions.R
</td>
<td style="text-align:left;">
code/plot_functions.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/main_cluster_marker_annotations.tsv
</td>
<td style="text-align:left;">
main_cluster_marker_annotations.tsv
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong668678e">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 668678e
</a>
</p>
</div>
<div id="strongRepositoryversionstrong668678e"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 668678e.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/Xenium_processing.Rmd
    Untracked:  analysis/annotation.Rmd
    Untracked:  analysis/post_clustering.Rmd
    Untracked:  analysis/splitting_samples.Rmd
    Untracked:  code/PIPAC_colors_themes.R
    Untracked:  code/anndata_to_seurat.R
    Untracked:  code/plot_functions.R
    Untracked:  code/rapids_pipeline/
    Untracked:  main_cluster_marker_annotations.tsv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="packages-and-environment-variables" class="section level3">
<h3>Packages and environment variables</h3>
<pre class="r"><code>suppressPackageStartupMessages({
  library(workflowr)
  library(arrow)
  library(Seurat)
  library(SeuratObject)
  library(SeuratDisk)
  library(tidyverse)
  library(tibble)
  library(ggplot2)
  library(ggpubr)
  library(ggrepel)
  library(googlesheets4)
  library(workflowr)})</code></pre>
</div>
<div id="environment-variables-and-helper-functions"
class="section level3">
<h3>Environment variables and helper functions</h3>
<pre class="r"><code>setwd(&quot;/home/hnatri/PIPAC_spatial/&quot;)
set.seed(9999)
options(scipen = 99999)
options(ggrepel.max.overlaps = Inf)

source(&quot;/home/hnatri/PIPAC_spatial/code/PIPAC_colors_themes.R&quot;)
source(&quot;/home/hnatri/PIPAC_spatial/code/plot_functions.R&quot;)</code></pre>
</div>
<div id="import-data" class="section level3">
<h3>Import data</h3>
<pre class="r"><code># Copied to isilon /tgen_labs/banovich/PIPAC/Seurat
seurat_data &lt;- readRDS(&quot;/scratch/hnatri/PIPAC/merged_spatial_filtered_split_clustered_NC50_NN20_PC20_Seurat.rds&quot;)
head(seurat_data@meta.data)</code></pre>
<pre><code>                                  orig.ident nCount_RNA nFeature_RNA
S21-24369_2A_TMA1_aaaafhkm-1_1 SeuratProject   16.17795           19
S21-7951_5A_TMA1_aaabeech-1_1  SeuratProject   46.16673           52
S21-7951_5A_TMA1_aaabidlm-1_1  SeuratProject   14.42256           19
S21-7951_5A_TMA1_aaacanmk-1_1  SeuratProject   15.59016           20
S21-7951_5A_TMA1_aaacnggd-1_1  SeuratProject   22.41628           28
S21-7951_5A_TMA1_aaacoafb-1_1  SeuratProject   37.49697           39
                                                      cell_id x_centroid
S21-24369_2A_TMA1_aaaafhkm-1_1 S21-24369_2A_TMA1_aaaafhkm-1_1   3120.635
S21-7951_5A_TMA1_aaabeech-1_1   S21-7951_5A_TMA1_aaabeech-1_1   3260.058
S21-7951_5A_TMA1_aaabidlm-1_1   S21-7951_5A_TMA1_aaabidlm-1_1   3167.581
S21-7951_5A_TMA1_aaacanmk-1_1   S21-7951_5A_TMA1_aaacanmk-1_1   3158.915
S21-7951_5A_TMA1_aaacnggd-1_1   S21-7951_5A_TMA1_aaacnggd-1_1   3176.202
S21-7951_5A_TMA1_aaacoafb-1_1   S21-7951_5A_TMA1_aaacoafb-1_1   3074.708
                               y_centroid transcript_counts
S21-24369_2A_TMA1_aaaafhkm-1_1   3213.421                34
S21-7951_5A_TMA1_aaabeech-1_1    5974.841               153
S21-7951_5A_TMA1_aaabidlm-1_1    6008.590                68
S21-7951_5A_TMA1_aaacanmk-1_1    6016.890                45
S21-7951_5A_TMA1_aaacnggd-1_1    6007.106               194
S21-7951_5A_TMA1_aaacoafb-1_1    5485.156                75
                               control_probe_counts genomic_control_counts
S21-24369_2A_TMA1_aaaafhkm-1_1                    0                      0
S21-7951_5A_TMA1_aaabeech-1_1                     0                      0
S21-7951_5A_TMA1_aaabidlm-1_1                     0                      0
S21-7951_5A_TMA1_aaacanmk-1_1                     0                      0
S21-7951_5A_TMA1_aaacnggd-1_1                     0                      0
S21-7951_5A_TMA1_aaacoafb-1_1                     0                      0
                               control_codeword_counts
S21-24369_2A_TMA1_aaaafhkm-1_1                       0
S21-7951_5A_TMA1_aaabeech-1_1                        0
S21-7951_5A_TMA1_aaabidlm-1_1                        0
S21-7951_5A_TMA1_aaacanmk-1_1                        0
S21-7951_5A_TMA1_aaacnggd-1_1                        0
S21-7951_5A_TMA1_aaacoafb-1_1                        0
                               unassigned_codeword_counts
S21-24369_2A_TMA1_aaaafhkm-1_1                          0
S21-7951_5A_TMA1_aaabeech-1_1                           0
S21-7951_5A_TMA1_aaabidlm-1_1                           0
S21-7951_5A_TMA1_aaacanmk-1_1                           0
S21-7951_5A_TMA1_aaacnggd-1_1                           0
S21-7951_5A_TMA1_aaacoafb-1_1                           0
                               deprecated_codeword_counts total_counts
S21-24369_2A_TMA1_aaaafhkm-1_1                          0           34
S21-7951_5A_TMA1_aaabeech-1_1                           0          153
S21-7951_5A_TMA1_aaabidlm-1_1                           0           68
S21-7951_5A_TMA1_aaacanmk-1_1                           0           45
S21-7951_5A_TMA1_aaacnggd-1_1                           0          194
S21-7951_5A_TMA1_aaacoafb-1_1                           0           75
                               cell_area nucleus_area nucleus_count
S21-24369_2A_TMA1_aaaafhkm-1_1  60.64485     33.68656             1
S21-7951_5A_TMA1_aaabeech-1_1  146.26110     32.06094             2
S21-7951_5A_TMA1_aaabidlm-1_1   75.36578     32.06094             1
S21-7951_5A_TMA1_aaacanmk-1_1   49.67188     18.78500             1
S21-7951_5A_TMA1_aaacnggd-1_1  126.75360     24.97141             1
S21-7951_5A_TMA1_aaacoafb-1_1   29.89344     14.90156             1
                                                                segmentation_method
S21-24369_2A_TMA1_aaaafhkm-1_1 Segmented by boundary stain (ATP1A1+CD45+E-Cadherin)
S21-7951_5A_TMA1_aaabeech-1_1  Segmented by boundary stain (ATP1A1+CD45+E-Cadherin)
S21-7951_5A_TMA1_aaabidlm-1_1  Segmented by boundary stain (ATP1A1+CD45+E-Cadherin)
S21-7951_5A_TMA1_aaacanmk-1_1  Segmented by boundary stain (ATP1A1+CD45+E-Cadherin)
S21-7951_5A_TMA1_aaacnggd-1_1  Segmented by boundary stain (ATP1A1+CD45+E-Cadherin)
S21-7951_5A_TMA1_aaacoafb-1_1  Segmented by boundary stain (ATP1A1+CD45+E-Cadherin)
                               num.blank           TMA percent.blank
S21-24369_2A_TMA1_aaaafhkm-1_1         0 MR_PIPAC-TMA1             0
S21-7951_5A_TMA1_aaabeech-1_1          0 MR_PIPAC-TMA1             0
S21-7951_5A_TMA1_aaabidlm-1_1          0 MR_PIPAC-TMA1             0
S21-7951_5A_TMA1_aaacanmk-1_1          0 MR_PIPAC-TMA1             0
S21-7951_5A_TMA1_aaacnggd-1_1          0 MR_PIPAC-TMA1             0
S21-7951_5A_TMA1_aaacoafb-1_1          0 MR_PIPAC-TMA1             0
                               nCount_cell_RNA nFeature_cell_RNA       Sample
S21-24369_2A_TMA1_aaaafhkm-1_1              34                22 S21-24369_2A
S21-7951_5A_TMA1_aaabeech-1_1              153                72  S21-7951_5A
S21-7951_5A_TMA1_aaabidlm-1_1               68                37  S21-7951_5A
S21-7951_5A_TMA1_aaacanmk-1_1               45                31  S21-7951_5A
S21-7951_5A_TMA1_aaacnggd-1_1              194                88  S21-7951_5A
S21-7951_5A_TMA1_aaacoafb-1_1               75                40  S21-7951_5A
                                       ident leiden_res0.5 leiden_res1.0
S21-24369_2A_TMA1_aaaafhkm-1_1 SeuratProject             2             6
S21-7951_5A_TMA1_aaabeech-1_1  SeuratProject            10             2
S21-7951_5A_TMA1_aaabidlm-1_1  SeuratProject             4             4
S21-7951_5A_TMA1_aaacanmk-1_1  SeuratProject             4            21
S21-7951_5A_TMA1_aaacnggd-1_1  SeuratProject            10             2
S21-7951_5A_TMA1_aaacoafb-1_1  SeuratProject             5             1
                               leiden_res1.5 leiden_res2.0 Institution
S21-24369_2A_TMA1_aaaafhkm-1_1             8            12         NWH
S21-7951_5A_TMA1_aaabeech-1_1             27            35         NWH
S21-7951_5A_TMA1_aaabidlm-1_1              2             8         NWH
S21-7951_5A_TMA1_aaacanmk-1_1              2            31         NWH
S21-7951_5A_TMA1_aaacnggd-1_1             27            35         NWH
S21-7951_5A_TMA1_aaacoafb-1_1             22            40         NWH
                                Patient_ID Timepoint Tissue Location_Quadrant
S21-24369_2A_TMA1_aaaafhkm-1_1 S21-NWH-012        12  Tumor       Right Lower
S21-7951_5A_TMA1_aaabeech-1_1  S21-NWH-011         0 Normal           Unknown
S21-7951_5A_TMA1_aaabidlm-1_1  S21-NWH-011         0 Normal           Unknown
S21-7951_5A_TMA1_aaacanmk-1_1  S21-NWH-011         0 Normal           Unknown
S21-7951_5A_TMA1_aaacnggd-1_1  S21-NWH-011         0 Normal           Unknown
S21-7951_5A_TMA1_aaacoafb-1_1  S21-NWH-011         0 Normal           Unknown</code></pre>
<pre class="r"><code>unique(seurat_data$leiden_res0.5)</code></pre>
<pre><code> [1] 2  10 4  5  1  3  9  11 0  7  8  6  12
Levels: 0 1 2 3 4 5 6 7 8 9 10 11 12</code></pre>
<pre class="r"><code>DimPlot(seurat_data,
        group.by = &quot;leiden_res0.5&quot;,
        cols = pipac_cluster_col,
        reduction = &quot;umap&quot;,
        raster = T,
        label = T) +
  coord_fixed(ratio = 1) +
  theme_minimal() +
  NoLegend()</code></pre>
<p><img src="figure/annotation.Rmd/unnamed-chunk-3-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="marker-information" class="section level3">
<h3>Marker information</h3>
<pre class="r"><code>gs4_deauth()
markers  &lt;- gs4_get(&quot;https://docs.google.com/spreadsheets/d/1sXXwOreLxjMSUoPt79c6jmaQpluWkaxA5P5HfDsed3I/edit?usp=sharing&quot;)
markers &lt;- read_sheet(markers, sheet = &quot;Markers&quot;)

head(markers)</code></pre>
<pre><code># A tibble: 6 × 8
  Gene   Ensemble_ID  Num_Probesets Codewords Annotation Annotation2 Annotation3
  &lt;chr&gt;  &lt;chr&gt;                &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt;      
1 CMKLR1 ENSG0000017…             8         1 Angiogene… Tumor       Tumor      
2 BCL2   ENSG0000017…             8         1 Apoptosis  &lt;NA&gt;        &lt;NA&gt;       
3 FADD   ENSG0000016…             7         1 Apoptosis… Tumor       Tumor      
4 LAMP2  ENSG0000000…             5         1 Autophagy  &lt;NA&gt;        &lt;NA&gt;       
5 ADAM28 ENSG0000004…             4         1 B Cell     Bcell       Lymphoid   
6 BANK1  ENSG0000015…             8         1 B Cell     Bcell       Lymphoid   
# ℹ 1 more variable: Annotation4 &lt;chr&gt;</code></pre>
<pre class="r"><code>table(markers$Annotation4)</code></pre>
<pre><code>
                 Cellcycle               Cytoskeleton 
                         4                          1 
               Cytoskelton                Endothelial 
                         1                          1 
                Epithelial           Epithelial;Tumor 
                         8                          3 
                    Immune          Immune;Epithelial 
                       198                          1 
Immune;Mesenchymal;Stromal           Immune;Signaling 
                         1                         16 
              Immune;Tumor                Mesenchymal 
                         1                          4 
        Mesenchymal;Immune                  Signaling 
                         1                         24 
                   Stromal                      Tumor 
                         2                         52 </code></pre>
</div>
<div id="top-cluster-markers" class="section level3">
<h3>Top cluster markers</h3>
<pre class="r"><code>Idents(seurat_data) &lt;- seurat_data$leiden_res0.5
cluster_markers &lt;- FindAllMarkers(seurat_data,
                                  return.thresh = 0.01,
                                  logfc.threshold = 0.5,
                                  min.pct = 0.20,
                                  verbose = T)

table(cluster_markers$cluster)</code></pre>
<pre><code>
  0   1   2   3   4   5   6   7   8   9  10  11  12 
 50  33 100  89  37 121  84  43  34  56  46  66  53 </code></pre>
<pre class="r"><code>hist(cluster_markers$avg_log2FC, main = &quot;&quot;, xlab = &quot;avg_log2FC&quot;, breaks = 100)</code></pre>
<p><img src="figure/annotation.Rmd/unnamed-chunk-5-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(cluster_markers$p_val, main = &quot;&quot;, xlab = &quot;p_val&quot;, breaks = 100)</code></pre>
<p><img src="figure/annotation.Rmd/unnamed-chunk-5-2.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(cluster_markers$p_val_adj, main = &quot;&quot;, xlab = &quot;p_val_adj&quot;, breaks = 100)</code></pre>
<p><img src="figure/annotation.Rmd/unnamed-chunk-5-3.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>top_cluster_markers &lt;- cluster_markers %&gt;%
  arrange(dplyr::desc(avg_log2FC)) %&gt;%
  group_by(cluster) %&gt;%
  dplyr::slice(1:10)</code></pre>
<pre class="r"><code>plot_features &lt;- c(&quot;PTPRC&quot;,
                   &quot;CD3D&quot;, &quot;CD3E&quot;, &quot;CD4&quot;, &quot;CD8A&quot;, # T cells
                   &quot;STAT4&quot;, &quot;STAT3&quot;, &quot;TIGIT&quot;, &quot;GZMB&quot;,
                   &quot;SELL&quot;, &quot;CD19&quot;, # B cells
                   &quot;CD68&quot;, &quot;CD44&quot;, &quot;MARCO&quot;, # Macrophages
                   &quot;C1QB&quot;, &quot;C1QBP&quot;,
                   &quot;MUC5AC&quot;, &quot;NOTCH3&quot;, &quot;MS4A1&quot;, &quot;PGA5&quot;, # Lineage markers
                   &quot;FN1&quot;, &quot;DCN&quot;, &quot;LUM&quot;, # Fibroblasts
                   &quot;EGR3&quot;, &quot;TP53&quot;, &quot;JUN&quot;, &quot;KIT&quot;, # Tumor
                   &quot;SOX9&quot;, &quot;RNF43&quot;)

DotPlot(seurat_data,
        group.by = &quot;leiden_res0.5&quot;,
        features = plot_features,
        cols = c(&quot;azure&quot;, &quot;tomato3&quot;)) +
  RotatedAxis()</code></pre>
<p><img src="figure/annotation.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>create_dotplot_heatmap(seurat_object = seurat_data,
                       plot_features = unique(top_cluster_markers$gene),
                       group_var = &quot;leiden_res0.5&quot;,
                       group_colors = pipac_cluster_col,
                       column_title = &quot;&quot;,
                       row_km = 5,
                       col_km = 5,
                       row.order = NULL,
                       col.order = NULL)</code></pre>
<p><img src="figure/annotation.Rmd/unnamed-chunk-7-1.png" width="576" style="display: block; margin: auto;" /><img src="figure/annotation.Rmd/unnamed-chunk-7-2.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="saving-top-markers-and-annotations" class="section level3">
<h3>Saving top markers and annotations</h3>
<pre class="r"><code>output_cluster_markers &lt;- cluster_markers %&gt;%
  arrange(dplyr::desc(avg_log2FC)) %&gt;%
  group_by(cluster) %&gt;%
  dplyr::slice(1:30)

output_cluster_markers &lt;- merge(top_cluster_markers, markers, by.x = &quot;gene&quot;, by.y = &quot;Gene&quot;)

write.table(output_cluster_markers, &quot;/home/hnatri/PIPAC_spatial/main_cluster_marker_annotations.tsv&quot;,
            quote = F, row.names = F, sep = &quot;\t&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ComplexHeatmap_2.18.0 viridis_0.6.3         viridisLite_0.4.2    
 [4] circlize_0.4.15       plyr_1.8.8            RColorBrewer_1.1-3   
 [7] googlesheets4_1.1.0   ggrepel_0.9.3         ggpubr_0.6.0         
[10] lubridate_1.9.2       forcats_1.0.0         stringr_1.5.0        
[13] dplyr_1.1.2           purrr_1.0.2           readr_2.1.4          
[16] tidyr_1.3.0           tibble_3.2.1          ggplot2_3.4.2        
[19] tidyverse_2.0.0       SeuratDisk_0.0.0.9021 Seurat_5.0.1         
[22] SeuratObject_5.0.2    sp_1.6-1              arrow_12.0.0         
[25] workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.20       splines_4.3.0          later_1.3.1           
  [4] cellranger_1.1.0       polyclip_1.10-4        fastDummies_1.7.3     
  [7] lifecycle_1.0.3        rstatix_0.7.2          doParallel_1.0.17     
 [10] rprojroot_2.0.3        globals_0.16.2         processx_3.8.1        
 [13] lattice_0.21-8         hdf5r_1.3.8            MASS_7.3-60           
 [16] backports_1.4.1        magrittr_2.0.3         limma_3.58.1          
 [19] plotly_4.10.2          sass_0.4.6             rmarkdown_2.22        
 [22] jquerylib_0.1.4        yaml_2.3.7             httpuv_1.6.11         
 [25] sctransform_0.4.1      spam_2.9-1             spatstat.sparse_3.0-1 
 [28] reticulate_1.29        cowplot_1.1.1          pbapply_1.7-0         
 [31] abind_1.4-5            Rtsne_0.16             presto_1.0.0          
 [34] BiocGenerics_0.48.1    git2r_0.32.0           S4Vectors_0.40.2      
 [37] IRanges_2.36.0         irlba_2.3.5.1          listenv_0.9.0         
 [40] spatstat.utils_3.0-3   goftest_1.2-3          RSpectra_0.16-1       
 [43] spatstat.random_3.1-5  fitdistrplus_1.1-11    parallelly_1.36.0     
 [46] leiden_0.4.3           codetools_0.2-19       tidyselect_1.2.0      
 [49] shape_1.4.6            farver_2.1.1           stats4_4.3.0          
 [52] matrixStats_1.0.0      spatstat.explore_3.2-1 googledrive_2.1.0     
 [55] jsonlite_1.8.5         GetoptLong_1.0.5       ellipsis_0.3.2        
 [58] progressr_0.13.0       iterators_1.0.14       ggridges_0.5.4        
 [61] survival_3.5-5         foreach_1.5.2          tools_4.3.0           
 [64] ica_1.0-3              Rcpp_1.0.10            glue_1.6.2            
 [67] gridExtra_2.3          xfun_0.39              withr_2.5.0           
 [70] fastmap_1.1.1          fansi_1.0.4            callr_3.7.3           
 [73] digest_0.6.31          timechange_0.2.0       R6_2.5.1              
 [76] mime_0.12              colorspace_2.1-0       Cairo_1.6-0           
 [79] scattermore_1.2        tensor_1.5             spatstat.data_3.0-1   
 [82] utf8_1.2.3             generics_0.1.3         data.table_1.14.8     
 [85] httr_1.4.6             htmlwidgets_1.6.2      whisker_0.4.1         
 [88] uwot_0.1.14            pkgconfig_2.0.3        gtable_0.3.3          
 [91] lmtest_0.9-40          htmltools_0.5.5        carData_3.0-5         
 [94] dotCall64_1.0-2        clue_0.3-64            scales_1.2.1          
 [97] png_0.1-8              knitr_1.43             rstudioapi_0.14       
[100] rjson_0.2.21           tzdb_0.4.0             reshape2_1.4.4        
[103] curl_5.0.1             nlme_3.1-162           cachem_1.0.8          
[106] zoo_1.8-12             GlobalOptions_0.1.2    KernSmooth_2.23-21    
[109] parallel_4.3.0         miniUI_0.1.1.1         pillar_1.9.0          
[112] vctrs_0.6.5            RANN_2.6.1             promises_1.2.0.1      
[115] car_3.1-2              xtable_1.8-4           cluster_2.1.4         
[118] evaluate_0.21          magick_2.7.4           cli_3.6.1             
[121] compiler_4.3.0         rlang_1.1.1            crayon_1.5.2          
[124] future.apply_1.11.0    ggsignif_0.6.4         labeling_0.4.2        
[127] ps_1.7.5               getPass_0.2-4          fs_1.6.2              
[130] stringi_1.7.12         deldir_1.0-9           assertthat_0.2.1      
[133] munsell_0.5.0          lazyeval_0.2.2         spatstat.geom_3.2-1   
[136] Matrix_1.6-5           RcppHNSW_0.5.0         hms_1.1.3             
[139] patchwork_1.1.2        bit64_4.0.5            future_1.32.0         
[142] statmod_1.5.0          shiny_1.7.4            highr_0.10            
[145] ROCR_1.0-11            gargle_1.4.0           igraph_1.4.3          
[148] broom_1.0.4            bslib_0.4.2            bit_4.0.5             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
