<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="heinin" />

<meta name="date" content="2025-03-19" />

<title>PIPAC Arm 3 comparative analysis</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIPAC_spatial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">PIPAC Arm 3 comparative analysis</h1>
<h4 class="author">heinin</h4>
<h4 class="date">2025-03-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>PIPAC_spatial/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240917code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240917)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240917code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240917)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/
</td>
<td style="text-align:left;">
.
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/code/PIPAC_colors_themes.R
</td>
<td style="text-align:left;">
code/PIPAC_colors_themes.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/code/plot_functions.R
</td>
<td style="text-align:left;">
code/plot_functions.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomheininPIPACspatialtree3bbda1950b67a2205d9bebb3344530d4fbf8bcbdtargetblank3bbda19a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/heinin/PIPAC_spatial/tree/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd" target="_blank">3bbda19</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomheininPIPACspatialtree3bbda1950b67a2205d9bebb3344530d4fbf8bcbdtargetblank3bbda19a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/heinin/PIPAC_spatial/tree/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd" target="_blank">3bbda19</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    analysis/figure/
    Ignored:    celltype_markers.tsv
    Ignored:    immune_cluster_marker_annotations.tsv
    Ignored:    immune_cluster_marker_annotations_2ndpass.tsv
    Ignored:    main_cluster_marker_annotations.tsv
    Ignored:    nonimmune_cluster_marker_annotations.tsv
    Ignored:    nonimmune_cluster_marker_annotations_2ndpass.tsv

Unstaged changes:
    Modified:   analysis/arm3_comparative_analysis.Rmd
    Modified:   analysis/niche_construction.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/arm3_comparative_analysis.Rmd</code>) and HTML
(<code>docs/arm3_comparative_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/analysis/arm3_comparative_analysis.Rmd" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
<td>
Updated annotations, added feature expression by celltype
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/heinin/PIPAC_spatial/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/arm3_comparative_analysis.html" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
<td>
Updated annotations, added feature expression by celltype
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>Load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(workflowr)
  library(arrow)
  library(Seurat)
  library(SeuratObject)
  library(SeuratDisk)
  library(tidyverse)
  library(tibble)
  library(ggplot2)
  library(ggpubr)
  library(ggrepel)
  library(googlesheets4)
  library(workflowr)
  library(patchwork)
  library(ggpmisc)
  library(scProportionTest)})</code></pre>
</div>
<div id="environment-variables-and-helper-functions"
class="section level2">
<h2>Environment variables and helper functions</h2>
<pre class="r"><code>setwd(&quot;/home/hnatri/PIPAC_spatial/&quot;)
set.seed(9999)
options(scipen = 99999)
options(ggrepel.max.overlaps = Inf)

source(&quot;/home/hnatri/PIPAC_spatial/code/PIPAC_colors_themes.R&quot;)
source(&quot;/home/hnatri/PIPAC_spatial/code/plot_functions.R&quot;)</code></pre>
</div>
<div id="marker-information" class="section level2">
<h2>Marker information</h2>
<pre class="r"><code>gs4_deauth()
metadata  &lt;- gs4_get(&quot;https://docs.google.com/spreadsheets/d/1sXXwOreLxjMSUoPt79c6jmaQpluWkaxA5P5HfDsed3I/edit?usp=sharing&quot;)
markers &lt;- read_sheet(metadata, sheet = &quot;Markers&quot;)
second_pass_nonimmune &lt;- read_sheet(metadata, sheet = &quot;Tumor 2nd pass&quot;)
second_pass_immune &lt;- read_sheet(metadata, sheet = &quot;Immune 2nd pass&quot;)

# Cell type markers by group
gs4_deauth()
markers_by_group  &lt;- gs4_get(&quot;https://docs.google.com/spreadsheets/d/1w0wrL-5KwNEWi_VpmriN7YmfwpBPlEYX0MKlwI2ZQu8/edit?usp=sharing&quot;)
markers_by_group &lt;- read_sheet(markers_by_group, sheet = &quot;Markers by group&quot;)</code></pre>
</div>
<div id="import-and-subset-data" class="section level2">
<h2>Import and subset data</h2>
<pre class="r"><code>seurat_data &lt;- readRDS(&quot;/tgen_labs/banovich/PIPAC/Seurat/PIPAC_NC50_NN20_PC20_Seurat_annotated_metadata_niches.rds&quot;)
DefaultAssay(seurat_data) &lt;- &quot;RNA&quot;
seurat_data &lt;- seurat_data %&gt;%
  NormalizeData() %&gt;%
  ScaleData()

seurat_arm3 &lt;- subset(seurat_data, subset = Arm == &quot;Arm3&quot;)
seurat_arm3 &lt;- seurat_arm3 %&gt;%
  NormalizeData() %&gt;%
  ScaleData()

length(unique(seurat_arm3$Patient_ID))</code></pre>
<pre><code>[1] 7</code></pre>
<pre class="r"><code>length(unique(seurat_arm3$Sample))</code></pre>
<pre><code>[1] 28</code></pre>
<pre class="r"><code>unique(seurat_arm3$Timepoint)</code></pre>
<pre><code>[1] &quot;6&quot; &quot;0&quot;</code></pre>
</div>
<div id="all-tumors" class="section level2">
<h2>All tumors</h2>
<div id="cell-type-proportions-by-sample-and-sample-type"
class="section level3">
<h3>Cell type proportions by sample and sample type</h3>
<pre class="r"><code>create_barplot(seurat_arm3,
               group_var = &quot;Sample&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(seurat_arm3$Annotation))),
               group_levels = sort(unique(seurat_arm3$Sample)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-5-1.png" width="2304" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-5-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-type-proportions-by-timepoint-in-tumors"
class="section level3">
<h3>Cell type proportions by timepoint in tumors</h3>
<pre class="r"><code>create_barplot(subset(seurat_arm3, subset = Tissue == &quot;Tumor&quot;),
               group_var = &quot;Timepoint&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(seurat_arm3$Annotation))),
               group_levels = sort(unique(seurat_arm3$Timepoint)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-6-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-type-proportions-by-timepoint-in-tumor-adjacent-samples"
class="section level3">
<h3>Cell type proportions by timepoint in tumor-adjacent samples</h3>
<pre class="r"><code>create_barplot(subset(seurat_arm3, subset = Tissue == &quot;Normal&quot;),
               group_var = &quot;Timepoint&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(seurat_arm3$Annotation))),
               group_levels = sort(unique(seurat_arm3$Timepoint)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-7-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-numbers-in-tumor-and-normal" class="section level3">
<h3>Cell numbers in tumor and normal</h3>
<pre class="r"><code>ct_table &lt;- as.data.frame(table(seurat_arm3$Annotation, seurat_arm3$Tissue))
colnames(ct_table) &lt;- c(&quot;Celltype&quot;, &quot;Tissue&quot;, &quot;Freq&quot;)
ct_table &lt;- spread(ct_table, Celltype, Freq)
# Converting to percetange 
#ct_table[,2:length(ct_table)] &lt;- (ct_table[,2:length(ct_table)]/rowSums(ct_table[,2:length(ct_table)]))*100
ct_table &lt;- gather(ct_table, Celltype, Freq, names(ct_table)[2:length(names(ct_table))], factor_key=TRUE)
    
ct_table$Tissue &lt;- factor(ct_table$Tissue, levels = c(&quot;Normal&quot;, &quot;Tumor&quot;))

ggplot(ct_table, aes(x = Tissue, y = Freq, fill = Celltype)) +
       geom_bar(stat=&quot;identity&quot;, position=&#39;stack&#39;, width = 0.8) +
       scale_fill_manual(name = &quot;Celltype&quot;, values = pipac_celltype_col) +
       xlab(&quot;&quot;) +
       ylab(&quot;Cell count&quot;) +
       theme_classic() +
       theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-8-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tumor-cell-type-proportions-by-response"
class="section level3">
<h3>Tumor cell type proportions by response</h3>
<pre class="r"><code>arm3_tumor &lt;- subset(seurat_arm3, subset = Tissue == &quot;Tumor&quot;)
arm3_tumor &lt;- arm3_tumor %&gt;%
  NormalizeData() %&gt;%
  ScaleData()

table(arm3_tumor$Patient_ID)</code></pre>
<pre><code>
COH-022 COH-038 COH-045 MFL-040 MFL-043 MFL-046 MFL-049 
   3129   11811    7838   16227   15649   22360   14658 </code></pre>
<pre class="r"><code>table(as.character(arm3_tumor$Sample))</code></pre>
<pre><code>
 20-010121T_(2_5_24) 20-010121T_(3_26_24)  20-010121T_(5_7_24) 
                6203                16157                 9158 
20-010121T_(6_19_24)             BSJ-38-1             BSJ-38-3 
                5500                 6315                 5496 
             JM-45-1              JM-45-3      SF230726-002257 
                5191                 2647                11105 
     SF230912-002063      SF230915-002249      SF231027-005404 
                5122                12437                 3212 
             SS-22-1              SS-22-3 
                2328                  801 </code></pre>
<pre class="r"><code>arm3_tumor@meta.data %&gt;%
  dplyr::select(c(&quot;Patient_ID&quot;, &quot;Sample&quot;)) %&gt;%
  distinct()</code></pre>
<pre><code>                                        Patient_ID               Sample
BSJ-38-3_TMA8_aaaalbkj-1_9                 COH-038             BSJ-38-3
SS-22-1_TMA8_aaagdmde-1_9                  COH-022              SS-22-1
20-010121T_(2_5_24)_TMA8_aacmgcaa-1_9      MFL-046  20-010121T_(2_5_24)
20-010121T_(6_19_24)_TMA8_aacodpjo-1_9     MFL-049 20-010121T_(6_19_24)
SF230726-002257_TMA8_abompjkc-1_9          MFL-040      SF230726-002257
SF231027-005404_TMA8_admleojl-1_9          MFL-043      SF231027-005404
JM-45-1_TMA9_aaabjfbe-1_10                 COH-045              JM-45-1
SS-22-3_TMA9_aagjmgfm-1_10                 COH-022              SS-22-3
BSJ-38-1_TMA9_aameglle-1_10                COH-038             BSJ-38-1
JM-45-3_TMA9_abehepnb-1_10                 COH-045              JM-45-3
20-010121T_(5_7_24)_TMA9_ablkgbhj-1_10     MFL-049  20-010121T_(5_7_24)
SF230915-002249_TMA9_accaokgb-1_10         MFL-043      SF230915-002249
SF230912-002063_TMA9_aecjangm-1_10         MFL-040      SF230912-002063
20-010121T_(3_26_24)_TMA9_aelaglgk-1_10    MFL-046 20-010121T_(3_26_24)</code></pre>
<pre class="r"><code>table(arm3_tumor$BestResponse)</code></pre>
<pre><code>
   PR    SD 
15649 37436 </code></pre>
<pre class="r"><code>create_barplot(arm3_tumor,
               group_var = &quot;BestResponse&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(arm3_tumor$Annotation))),
               group_levels = sort(unique(arm3_tumor$BestResponse)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-9-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="testing-for-significance-with-scproportiontest"
class="section level4">
<h4>Testing for significance with scProportionTest</h4>
<pre class="r"><code>prop_test &lt;- sc_utils(arm3_tumor)
prop_test &lt;- permutation_test(
    prop_test, cluster_identity = &quot;Annotation&quot;,
    sample_1 = &quot;PR&quot;, sample_2 = &quot;SD&quot;,
    sample_identity = &quot;BestResponse&quot;)

permutation_plot(prop_test) +
  ggtitle(&quot;All tumors&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-10-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="tumor-adjacent-cell-type-proportions-by-response"
class="section level3">
<h3>Tumor-adjacent cell type proportions by response</h3>
<pre class="r"><code>arm3_normal &lt;- subset(seurat_arm3, subset = Tissue == &quot;Normal&quot;)
arm3_normal &lt;- arm3_normal %&gt;%
  NormalizeData() %&gt;%
  ScaleData()

table(arm3_normal$Patient_ID)</code></pre>
<pre><code>
COH-022 COH-038 COH-045 MFL-040 MFL-043 MFL-046 MFL-049 
   6599    5515    6967   11290    8315    6499    7248 </code></pre>
<pre class="r"><code>table(as.character(arm3_normal$Sample))</code></pre>
<pre><code>
 20-010121N_(2_5_24) 20-010121N_(3_26_24)  20-010121N_(5_7_24) 
                3859                 2640                 2463 
20-010121N_(6_19_24)             BSJ-38-2             BSJ-38-4 
                4785                 2395                 3120 
             JM-45-2              JM-45-4      SF230726-002256 
                2877                 4090                 4156 
     SF230912-002062      SF230915-002250      SF231027-005405 
                7134                 3655                 4660 
             SS-22-2              SS-22-4 
                2878                 3721 </code></pre>
<pre class="r"><code>arm3_normal@meta.data %&gt;%
  dplyr::select(c(&quot;Patient_ID&quot;, &quot;Sample&quot;)) %&gt;%
  distinct()</code></pre>
<pre><code>                                        Patient_ID               Sample
20-010121N_(3_26_24)_TMA8_aaacehok-1_9     MFL-046 20-010121N_(3_26_24)
SF230912-002062_TMA8_aagpdaej-1_9          MFL-040      SF230912-002062
SF230915-002250_TMA8_aapkmjjf-1_9          MFL-043      SF230915-002250
BSJ-38-2_TMA8_abgdjnnp-1_9                 COH-038             BSJ-38-2
JM-45-4_TMA8_aedpaehf-1_9                  COH-045              JM-45-4
20-010121N_(5_7_24)_TMA8_aeecfihf-1_9      MFL-049  20-010121N_(5_7_24)
SS-22-4_TMA8_aeeklhem-1_9                  COH-022              SS-22-4
BSJ-38-4_TMA9_aaagjppb-1_10                COH-038             BSJ-38-4
JM-45-2_TMA9_aamalbcp-1_10                 COH-045              JM-45-2
SS-22-2_TMA9_abfhegoa-1_10                 COH-022              SS-22-2
20-010121N_(6_19_24)_TMA9_abobgfmj-1_10    MFL-049 20-010121N_(6_19_24)
SF231027-005405_TMA9_abokfaoo-1_10         MFL-043      SF231027-005405
20-010121N_(2_5_24)_TMA9_acinkijl-1_10     MFL-046  20-010121N_(2_5_24)
SF230726-002256_TMA9_afbnfahn-1_10         MFL-040      SF230726-002256</code></pre>
<pre class="r"><code>table(arm3_normal$BestResponse)</code></pre>
<pre><code>
   PR    SD 
 8315 26329 </code></pre>
<pre class="r"><code>create_barplot(arm3_normal,
               group_var = &quot;BestResponse&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(arm3_tumor$Annotation))),
               group_levels = sort(unique(arm3_tumor$BestResponse)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-11-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-11-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="testing-for-significance-with-scproportiontest-1"
class="section level4">
<h4>Testing for significance with scProportionTest</h4>
<pre class="r"><code>prop_test &lt;- sc_utils(arm3_normal)
prop_test &lt;- permutation_test(
    prop_test, cluster_identity = &quot;Annotation&quot;,
    sample_1 = &quot;PR&quot;, sample_2 = &quot;SD&quot;,
    sample_identity = &quot;BestResponse&quot;)

permutation_plot(prop_test) +
  ggtitle(&quot;All tumors&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-12-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cell-type-proportion-correlations-with-survival-time"
class="section level3">
<h3>Cell type proportion correlations with survival time</h3>
<p>Linear regression</p>
<pre class="r"><code>metadata &lt;- seurat_arm3@meta.data %&gt;%
  dplyr::select(&quot;Patient_ID&quot;, &quot;DISEASESITE&quot;, &quot;osmonths&quot;, &quot;pfsmonths&quot;, &quot;No_of_PIPACs&quot;) %&gt;%
  distinct()

prop_table &lt;- as.data.frame(table(seurat_arm3@meta.data$Annotation, as.character(seurat_arm3@meta.data$Patient_ID)))
colnames(prop_table) &lt;- c(&quot;Annotation&quot;, &quot;Patient_ID&quot;, &quot;Freq&quot;)
prop_table &lt;- spread(prop_table, Annotation, Freq)
# Converting to percentage
prop_table[,2:length(prop_table)] &lt;- (prop_table[,2:length(prop_table)]/rowSums(prop_table[,2:length(prop_table)]))*100
prop_table &lt;- gather(prop_table, Annotation, Freq, names(prop_table)[2:length(names(prop_table))], factor_key=TRUE)

# Adding survival data
prop_survival &lt;- merge(prop_table, metadata, by = &quot;Patient_ID&quot;)
prop_survival$osmonths &lt;- unlist(prop_survival$osmonths)
prop_survival$pfmonths &lt;- unlist(prop_survival$pfmonths)

# Plotting
plot_list &lt;- lapply(sort(unique(seurat_arm3$Annotation)), function(ct){
  prop_survival %&gt;%
    filter(Annotation == ct) %&gt;%
    ggplot(aes(x = Freq, y = osmonths, color = Annotation)) +
    geom_point() +
    scale_color_manual(name = &quot;Annotation&quot;, values = pipac_celltype_col) +
    stat_poly_line() +
    stat_poly_eq(formula = y ~ x,
                 rr.digits = 3, p.digits = 3, small.p = T,
                 aes(label = paste(after_stat(rr.label), after_stat(p.value.label), sep = &quot;*`,`~&quot;)),
                 label.x = 0.01, label.y = 1, parse = T) +
    ylab(&quot;Overall Survival&quot;) +
    xlab(paste0(&quot;% of &quot;, ct, &quot; cells&quot;)) +
    theme_classic() +
    theme(axis.text.x = element_text(color = &quot;black&quot;),
          axis.text.y = element_text(color = &quot;black&quot;)) +
    NoLegend()
})

wrap_plots(plot_list, ncol = 3)

names(plot_list) &lt;- sort(unique(seurat_arm3$Annotation))</code></pre>
</div>
<div id="fibroblast-marker-expression" class="section level3">
<h3>Fibroblast marker expression</h3>
<pre class="r"><code>DotPlot(subset(seurat_arm3, subset = Annotation %in% paste0(&quot;F&quot;, seq(1,5))),
        group.by = &quot;Annotation&quot;,
        features = unique(markers_by_group[which(markers_by_group$annotation1 == &quot;CAFs&quot;),]$feature),
        cols = c(&quot;azure&quot;, &quot;tomato3&quot;)) +
  coord_flip() +
  ggtitle(&quot;CAFs, all timepoints&quot;)</code></pre>
</div>
</div>
<div id="baseline-tumors" class="section level2">
<h2>Baseline tumors</h2>
<div id="cell-type-proportions-by-sample" class="section level3">
<h3>Cell type proportions by sample</h3>
<pre class="r"><code>unique(seurat_arm3$Timepoint)</code></pre>
<pre><code>[1] &quot;6&quot; &quot;0&quot;</code></pre>
<pre class="r"><code>baseline_tumor &lt;- subset(seurat_arm3, subset = Tissue == &quot;Tumor&quot; &amp;
                           Timepoint == 0)
baseline_tumor &lt;- baseline_tumor %&gt;%
  NormalizeData() %&gt;%
  ScaleData()

table(baseline_tumor$Patient_ID)</code></pre>
<pre><code>
COH-022 COH-038 COH-045 MFL-040 MFL-043 MFL-046 MFL-049 
   2328    6315    5191   11105   12437    6203    9158 </code></pre>
<pre class="r"><code>table(as.character(baseline_tumor$Sample))</code></pre>
<pre><code>
20-010121T_(2_5_24) 20-010121T_(5_7_24)            BSJ-38-1             JM-45-1 
               6203                9158                6315                5191 
    SF230726-002257     SF230915-002249             SS-22-1 
              11105               12437                2328 </code></pre>
<pre class="r"><code>baseline_tumor@meta.data %&gt;%
  dplyr::select(c(&quot;Patient_ID&quot;, &quot;Sample&quot;)) %&gt;%
  distinct()</code></pre>
<pre><code>                                       Patient_ID              Sample
SS-22-1_TMA8_aaagdmde-1_9                 COH-022             SS-22-1
20-010121T_(2_5_24)_TMA8_aacmgcaa-1_9     MFL-046 20-010121T_(2_5_24)
SF230726-002257_TMA8_abompjkc-1_9         MFL-040     SF230726-002257
JM-45-1_TMA9_aaabjfbe-1_10                COH-045             JM-45-1
BSJ-38-1_TMA9_aameglle-1_10               COH-038            BSJ-38-1
20-010121T_(5_7_24)_TMA9_ablkgbhj-1_10    MFL-049 20-010121T_(5_7_24)
SF230915-002249_TMA9_accaokgb-1_10        MFL-043     SF230915-002249</code></pre>
<pre class="r"><code>table(baseline_tumor$BestResponse)</code></pre>
<pre><code>
   PR    SD 
12437 22992 </code></pre>
<pre class="r"><code>create_barplot(baseline_tumor,
               group_var = &quot;Sample&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(baseline_tumor$Annotation))),
               group_levels = sort(unique(baseline_tumor$Sample)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-15-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-15-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-type-proportions-by-response" class="section level3">
<h3>Cell type proportions by response</h3>
<pre class="r"><code>create_barplot(baseline_tumor,
               group_var = &quot;BestResponse&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(baseline_tumor$Annotation))),
               group_levels = sort(unique(baseline_tumor$BestResponse)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-16-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-16-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="testing-for-significance-with-scproportiontest-2"
class="section level4">
<h4>Testing for significance with scProportionTest</h4>
<pre class="r"><code>prop_test &lt;- sc_utils(baseline_tumor)
prop_test &lt;- permutation_test(
    prop_test, cluster_identity = &quot;Annotation&quot;,
    sample_1 = &quot;PR&quot;, sample_2 = &quot;SD&quot;,
    sample_identity = &quot;BestResponse&quot;)

permutation_plot(prop_test) +
  ggtitle(&quot;Baseline tumors&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-17-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-17-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cell-type-proportions-by-response-in-tumor-adjacent"
class="section level3">
<h3>Cell type proportions by response in tumor-adjacent</h3>
<pre class="r"><code>baseline_normal &lt;- subset(seurat_arm3, subset = Tissue == &quot;Normal&quot; &amp;
                           Timepoint == 0)
baseline_normal &lt;- baseline_normal %&gt;%
  NormalizeData() %&gt;%
  ScaleData()

table(baseline_normal$Patient_ID)</code></pre>
<pre><code>
COH-022 COH-038 COH-045 MFL-040 MFL-043 MFL-046 MFL-049 
   2878    2395    2877    4156    3655    3859    2463 </code></pre>
<pre class="r"><code>table(as.character(baseline_normal$Sample))</code></pre>
<pre><code>
20-010121N_(2_5_24) 20-010121N_(5_7_24)            BSJ-38-2             JM-45-2 
               3859                2463                2395                2877 
    SF230726-002256     SF230915-002250             SS-22-2 
               4156                3655                2878 </code></pre>
<pre class="r"><code>baseline_normal@meta.data %&gt;%
  dplyr::select(c(&quot;Patient_ID&quot;, &quot;Sample&quot;)) %&gt;%
  distinct()</code></pre>
<pre><code>                                       Patient_ID              Sample
SF230915-002250_TMA8_aapkmjjf-1_9         MFL-043     SF230915-002250
BSJ-38-2_TMA8_abgdjnnp-1_9                COH-038            BSJ-38-2
20-010121N_(5_7_24)_TMA8_aeecfihf-1_9     MFL-049 20-010121N_(5_7_24)
JM-45-2_TMA9_aamalbcp-1_10                COH-045             JM-45-2
SS-22-2_TMA9_abfhegoa-1_10                COH-022             SS-22-2
20-010121N_(2_5_24)_TMA9_acinkijl-1_10    MFL-046 20-010121N_(2_5_24)
SF230726-002256_TMA9_afbnfahn-1_10        MFL-040     SF230726-002256</code></pre>
<pre class="r"><code>table(baseline_normal$BestResponse)</code></pre>
<pre><code>
   PR    SD 
 3655 10613 </code></pre>
<pre class="r"><code>table(baseline_normal$Annotation)</code></pre>
<pre><code>
       B1        B2        DC     Endo1     Endo2     Endo3 EpiTumor1        F1 
      108        12        66      4930       246       945        27      2463 
       F2        F3        F4        L1        L2        L3        L4        L5 
      169      3633        51         5       346       227       255         2 
       M1        M2        M4        M5        M6        M7        M8      Mast 
      490        75        83        61       250        29       520       106 
     Meso      Peri   Plasma1   Plasma2   Plasma3  Stromal1  Stromal2 
     1279      2908         4        68         8       354      2563 </code></pre>
<pre class="r"><code>create_barplot(baseline_normal,
               group_var = &quot;BestResponse&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(baseline_normal$Annotation))),
               group_levels = sort(unique(baseline_normal$BestResponse)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-18-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-18-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="testing-for-significance-with-scproportiontest-3"
class="section level4">
<h4>Testing for significance with scProportionTest</h4>
<pre class="r"><code>prop_test &lt;- sc_utils(baseline_normal)
prop_test &lt;- permutation_test(
    prop_test, cluster_identity = &quot;Annotation&quot;,
    sample_1 = &quot;PR&quot;, sample_2 = &quot;SD&quot;,
    sample_identity = &quot;BestResponse&quot;)

permutation_plot(prop_test) +
  ggtitle(&quot;Baseline tumor-adjacent&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-19-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-19-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="tumor-cell-type-proportion-correlations-with-survival-time"
class="section level3">
<h3>Tumor cell type proportion correlations with survival time</h3>
<p>Linear regression</p>
<pre class="r"><code>metadata &lt;- baseline_tumor@meta.data %&gt;%
  dplyr::select(&quot;Patient_ID&quot;, &quot;DISEASESITE&quot;, &quot;osmonths&quot;, &quot;pfsmonths&quot;, &quot;No_of_PIPACs&quot;) %&gt;%
  distinct()

prop_table &lt;- as.data.frame(table(baseline_tumor@meta.data$Annotation, as.character(baseline_tumor@meta.data$Patient_ID)))
colnames(prop_table) &lt;- c(&quot;Annotation&quot;, &quot;Patient_ID&quot;, &quot;Freq&quot;)
prop_table &lt;- spread(prop_table, Annotation, Freq)
# Converting to percentage
prop_table[,2:length(prop_table)] &lt;- (prop_table[,2:length(prop_table)]/rowSums(prop_table[,2:length(prop_table)]))*100
prop_table &lt;- gather(prop_table, Annotation, Freq, names(prop_table)[2:length(names(prop_table))], factor_key=TRUE)

# Adding survival data
prop_survival &lt;- merge(prop_table, metadata, by = &quot;Patient_ID&quot;)
prop_survival$osmonths &lt;- unlist(prop_survival$osmonths)
prop_survival$pfmonths &lt;- unlist(prop_survival$pfmonths)

# Plotting
plot_list &lt;- lapply(sort(unique(baseline_tumor$Annotation)), function(ct){
  prop_survival %&gt;%
    filter(Annotation == ct) %&gt;%
    ggplot(aes(x = Freq, y = osmonths, color = Annotation)) +
    geom_point() +
    scale_color_manual(name = &quot;Annotation&quot;, values = pipac_celltype_col) +
    stat_poly_line() +
    stat_poly_eq(formula = y ~ x,
                 rr.digits = 3, p.digits = 3, small.p = T,
                 aes(label = paste(after_stat(rr.label), after_stat(p.value.label), sep = &quot;*`,`~&quot;)),
                 label.x = 0.01, label.y = 1, parse = T) +
    ylab(&quot;Overall Survival&quot;) +
    xlab(paste0(&quot;% of &quot;, ct, &quot; cells&quot;)) +
    theme_classic() +
    theme(axis.text.x = element_text(color = &quot;black&quot;),
          axis.text.y = element_text(color = &quot;black&quot;)) +
    NoLegend()
})

wrap_plots(plot_list, ncol = 3)</code></pre>
</div>
</div>
<div id="after-treatment-tumor" class="section level2">
<h2>After treatment tumor</h2>
<div id="cell-type-proportions-by-sample-1" class="section level3">
<h3>Cell type proportions by sample</h3>
<pre class="r"><code>unique(seurat_arm3$Timepoint)</code></pre>
<pre><code>[1] &quot;6&quot; &quot;0&quot;</code></pre>
<pre class="r"><code>post_treatment_tumor &lt;- subset(seurat_arm3, subset = Tissue == &quot;Tumor&quot; &amp;
                                 Timepoint %in% c(6, 12))
post_treatment_tumor &lt;- post_treatment_tumor %&gt;%
  NormalizeData() %&gt;%
  ScaleData()

table(post_treatment_tumor$Patient_ID)</code></pre>
<pre><code>
COH-022 COH-038 COH-045 MFL-040 MFL-043 MFL-046 MFL-049 
    801    5496    2647    5122    3212   16157    5500 </code></pre>
<pre class="r"><code>table(as.character(post_treatment_tumor$Sample))</code></pre>
<pre><code>
20-010121T_(3_26_24) 20-010121T_(6_19_24)             BSJ-38-3 
               16157                 5500                 5496 
             JM-45-3      SF230912-002063      SF231027-005404 
                2647                 5122                 3212 
             SS-22-3 
                 801 </code></pre>
<pre class="r"><code>post_treatment_tumor@meta.data %&gt;%
  dplyr::select(c(&quot;Patient_ID&quot;, &quot;Sample&quot;)) %&gt;%
  distinct()</code></pre>
<pre><code>                                        Patient_ID               Sample
BSJ-38-3_TMA8_aaaalbkj-1_9                 COH-038             BSJ-38-3
20-010121T_(6_19_24)_TMA8_aacodpjo-1_9     MFL-049 20-010121T_(6_19_24)
SF231027-005404_TMA8_admleojl-1_9          MFL-043      SF231027-005404
SS-22-3_TMA9_aagjmgfm-1_10                 COH-022              SS-22-3
JM-45-3_TMA9_abehepnb-1_10                 COH-045              JM-45-3
SF230912-002063_TMA9_aecjangm-1_10         MFL-040      SF230912-002063
20-010121T_(3_26_24)_TMA9_aelaglgk-1_10    MFL-046 20-010121T_(3_26_24)</code></pre>
<pre class="r"><code>table(post_treatment_tumor$BestResponse)</code></pre>
<pre><code>
   PR    SD 
 3212 14444 </code></pre>
<pre class="r"><code>create_barplot(post_treatment_tumor,
               group_var = &quot;Sample&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(post_treatment_tumor$Annotation))),
               group_levels = sort(unique(post_treatment_tumor$Sample)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-21-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-21-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-type-proportions-by-response-1" class="section level3">
<h3>Cell type proportions by response</h3>
<pre class="r"><code>create_barplot(post_treatment_tumor,
               group_var = &quot;BestResponse&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(post_treatment_tumor$Annotation))),
               group_levels = sort(unique(post_treatment_tumor$BestResponse)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-22-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-22-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="testing-for-significance-with-scproportiontest-4"
class="section level4">
<h4>Testing for significance with scProportionTest</h4>
<pre class="r"><code>prop_test &lt;- sc_utils(post_treatment_tumor)
prop_test &lt;- permutation_test(
    prop_test, cluster_identity = &quot;Annotation&quot;,
    sample_1 = &quot;PR&quot;, sample_2 = &quot;SD&quot;,
    sample_identity = &quot;BestResponse&quot;)

permutation_plot(prop_test) +
  ggtitle(&quot;Post-treatment tumors&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-23-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-23-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cell-type-proportions-by-response-in-tumor-adjacent-1"
class="section level3">
<h3>Cell type proportions by response in tumor-adjacent</h3>
<pre class="r"><code>unique(seurat_arm3$Timepoint)</code></pre>
<pre><code>[1] &quot;6&quot; &quot;0&quot;</code></pre>
<pre class="r"><code>treated_normal &lt;- subset(seurat_arm3, subset = Tissue == &quot;Normal&quot; &amp;
                           Timepoint == c(6, 12))
treated_normal &lt;- treated_normal %&gt;%
  NormalizeData() %&gt;%
  ScaleData()

table(treated_normal$Patient_ID)</code></pre>
<pre><code>
COH-022 COH-038 COH-045 MFL-040 MFL-043 MFL-046 MFL-049 
   1859    1558    2047    3570    2328    1320    2395 </code></pre>
<pre class="r"><code>table(as.character(treated_normal$Sample))</code></pre>
<pre><code>
20-010121N_(3_26_24) 20-010121N_(6_19_24)             BSJ-38-4 
                1320                 2395                 1558 
             JM-45-4      SF230912-002062      SF231027-005405 
                2047                 3570                 2328 
             SS-22-4 
                1859 </code></pre>
<pre class="r"><code>treated_normal@meta.data %&gt;%
  dplyr::select(c(&quot;Patient_ID&quot;, &quot;Sample&quot;)) %&gt;%
  distinct()</code></pre>
<pre><code>                                        Patient_ID               Sample
20-010121N_(3_26_24)_TMA8_aaacehok-1_9     MFL-046 20-010121N_(3_26_24)
SF230912-002062_TMA8_aahccjbj-1_9          MFL-040      SF230912-002062
JM-45-4_TMA8_aedpaehf-1_9                  COH-045              JM-45-4
SS-22-4_TMA8_aeeklhem-1_9                  COH-022              SS-22-4
BSJ-38-4_TMA9_aaagjppb-1_10                COH-038             BSJ-38-4
20-010121N_(6_19_24)_TMA9_abocijcg-1_10    MFL-049 20-010121N_(6_19_24)
SF231027-005405_TMA9_abokfaoo-1_10         MFL-043      SF231027-005405</code></pre>
<pre class="r"><code>table(treated_normal$BestResponse)</code></pre>
<pre><code>
  PR   SD 
2328 7859 </code></pre>
<pre class="r"><code>table(treated_normal$Annotation)</code></pre>
<pre><code>
       B1        B2        DC     Endo1     Endo2     Endo3 EpiTumor1        F1 
      161        19        41      1573       240      1105        13      1107 
       F2        F3        F4        L1        L2        L3        L4        L5 
      259      3054        20        24       337       385       123        13 
       M1        M2        M4        M5        M6        M7        M8      Mast 
      599       331       264        95       288        17      1523       123 
     Meso      Peri   Plasma1   Plasma2   Plasma3  Stromal1  Stromal2 
      559      1268        21        61        10       211      1233 </code></pre>
<pre class="r"><code>create_barplot(treated_normal,
               group_var = &quot;BestResponse&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(treated_normal$Annotation))),
               group_levels = sort(unique(treated_normal$BestResponse)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-24-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-24-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="testing-for-significance-with-scproportiontest-5"
class="section level4">
<h4>Testing for significance with scProportionTest</h4>
<pre class="r"><code>prop_test &lt;- sc_utils(treated_normal)
prop_test &lt;- permutation_test(
    prop_test, cluster_identity = &quot;Annotation&quot;,
    sample_1 = &quot;PR&quot;, sample_2 = &quot;SD&quot;,
    sample_identity = &quot;BestResponse&quot;)

permutation_plot(prop_test) +
  ggtitle(&quot;Post-treatment tumor-adjacent&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-25-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-25-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tumor-cell-type-proportion-correlations-with-survival-time-1"
class="section level4">
<h4>Tumor cell type proportion correlations with survival time</h4>
<p>Linear regression</p>
<pre class="r"><code>metadata &lt;- post_treatment_tumor@meta.data %&gt;%
  dplyr::select(&quot;Patient_ID&quot;, &quot;DISEASESITE&quot;, &quot;osmonths&quot;, &quot;pfsmonths&quot;, &quot;No_of_PIPACs&quot;) %&gt;%
  distinct()

prop_table &lt;- as.data.frame(table(post_treatment_tumor@meta.data$Annotation, as.character(post_treatment_tumor@meta.data$Patient_ID)))
colnames(prop_table) &lt;- c(&quot;Annotation&quot;, &quot;Patient_ID&quot;, &quot;Freq&quot;)
prop_table &lt;- spread(prop_table, Annotation, Freq)
# Converting to percentage
prop_table[,2:length(prop_table)] &lt;- (prop_table[,2:length(prop_table)]/rowSums(prop_table[,2:length(prop_table)]))*100
prop_table &lt;- gather(prop_table, Annotation, Freq, names(prop_table)[2:length(names(prop_table))], factor_key=TRUE)

# Adding survival data
prop_survival &lt;- merge(prop_table, metadata, by = &quot;Patient_ID&quot;)
prop_survival$osmonths &lt;- unlist(prop_survival$osmonths)
prop_survival$pfmonths &lt;- unlist(prop_survival$pfmonths)

# Plotting
plot_list &lt;- lapply(sort(unique(post_treatment_tumor$Annotation)), function(ct){
  prop_survival %&gt;%
    filter(Annotation == ct) %&gt;%
    ggplot(aes(x = Freq, y = osmonths, color = Annotation)) +
    geom_point() +
    scale_color_manual(name = &quot;Annotation&quot;, values = pipac_celltype_col) +
    stat_poly_line() +
    stat_poly_eq(formula = y ~ x,
                 rr.digits = 3, p.digits = 3, small.p = T,
                 aes(label = paste(after_stat(rr.label), after_stat(p.value.label), sep = &quot;*`,`~&quot;)),
                 label.x = 0.01, label.y = 1, parse = T) +
    ylab(&quot;Overall Survival&quot;) +
    xlab(paste0(&quot;% of &quot;, ct, &quot; cells&quot;)) +
    theme_classic() +
    theme(axis.text.x = element_text(color = &quot;black&quot;),
          axis.text.y = element_text(color = &quot;black&quot;)) +
    NoLegend()
})

wrap_plots(plot_list, ncol = 3)

names(plot_list) &lt;- sort(unique(post_treatment_tumor$Annotation))</code></pre>
<pre class="r"><code>plot_list[[&quot;F5&quot;]]</code></pre>
<p>Calculating Hazard ratios</p>
<pre class="r"><code>prop_table &lt;- as.data.frame(table(immune_fibro@meta.data$celltype, as.character(immune_fibro@meta.data$UPN)))
colnames(prop_table) &lt;- c(&quot;celltype&quot;, &quot;UPN&quot;, &quot;Freq&quot;)
prop_table &lt;- spread(prop_table, celltype, Freq)
# Converting to percentage
prop_table[,2:length(prop_table)] &lt;- (prop_table[,2:length(prop_table)]/rowSums(prop_table[,2:length(prop_table)]))*100
prop_table &lt;- gather(prop_table, celltype, Freq, names(prop_table)[2:length(names(prop_table))], factor_key=TRUE)

prop_table &lt;- prop_table %&gt;% pivot_wider(names_from = celltype,
                                         values_from = Freq)

# Adding survival data
metadata &lt;- immune_fibro@meta.data %&gt;%
  dplyr::select(&quot;UPN&quot;,
                &quot;Death.Status&quot;,
                &quot;Survival.time.in.Months.from.surgery&quot;,
                &quot;binary_response&quot;,
                &quot;Grade&quot;,
                &quot;Diagnosis.Histology&quot;,
                &quot;EGFR.Amplification&quot;,
                &quot;EGFRvIII..Exon.2.7.deletion&quot;,
                &quot;EGFR.Missense&quot;,
                &quot;TP53.Frameshift&quot;,
                &quot;TP53.Missense&quot;,
                &quot;TP53.Nonsense&quot;,
                &quot;IDH1.Missense&quot;,
                &quot;IDH2.Missense&quot;,
                &quot;TERT.Promoter.Mutation&quot;,
                &quot;PTEN.Codon.Deletion&quot;,
                &quot;PTEN.Frameshift&quot;,
                &quot;PTEN.Interference.of.splice.acceptor.site.in.Intron.6&quot;) %&gt;%
  distinct()

prop_survival &lt;- merge(prop_table, metadata, by = &quot;UPN&quot;)

# Fit survival data using the Kaplan-Meier method
surv_object &lt;- Surv(time = prop_survival$Survival.time.in.Months.from.surgery,
                    event = prop_survival$Death.Status)
surv_object 

# Fit a Cox proportional hazards model for a single gene
fit.coxph &lt;- coxph(surv_object ~ M1 + M2 + M3 + M4 + M5 + M6 + M7 + M8 + M9 + N1 +
                     L1 + L2 + L3 + L4 + L5 + L6 + L7 + L8 + L8 + L9 + L10 + B1 +
                     F1 + F2 + F3,
                   data = prop_survival)

p1 &lt;- ggforest(fit.coxph, data = prop_survival)

p1</code></pre>
</div>
</div>
<div id="inspecting-f5" class="section level3">
<h3>Inspecting F5</h3>
<div id="fibroblast-marker-expression-1" class="section level4">
<h4>Fibroblast marker expression</h4>
<pre class="r"><code>DotPlot(subset(post_treatment_tumor, subset = Annotation %in% paste0(&quot;F&quot;, seq(1,5))),
        group.by = &quot;Annotation&quot;,
        features = c(unique(markers_by_group[which(markers_by_group$annotation1 == &quot;CAFs&quot;),]$feature), &quot;DCN&quot;, &quot;LUM&quot;, &quot;EPCAM&quot;),
        cols = c(&quot;gray88&quot;, &quot;tomato3&quot;)) +
  coord_flip() +
  ggtitle(&quot;CAFs, post-treatment (D6/12)&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-29-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-29-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Idents(seurat_data) &lt;- seurat_data$Annotation
#celltype_markers &lt;- FindAllMarkers(seurat_data,
#                                   return.thresh = 0.01,
#                                   logfc.threshold = 0.5,
#                                   min.pct = 0.20,
#                                   only.pos = T,
#                                   verbose = F)
#
#table(celltype_markers$cluster)
#
#output_celltype_markers &lt;- celltype_markers %&gt;%
#  arrange(dplyr::desc(abs(avg_log2FC))) %&gt;%
#  group_by(cluster) %&gt;%
#  dplyr::slice(1:30)
#
#output_celltype_markers &lt;- merge(output_celltype_markers, markers, by.x = &quot;gene&quot;, by.y = &quot;Gene&quot;)</code></pre>
</div>
</div>
</div>
<div
id="tumor-cell-type-proportion-differences-between-timepoints-before-and-after-treatment"
class="section level2">
<h2>Tumor cell type proportion differences between timepoints (before
and after treatment)</h2>
<pre class="r"><code>arm3_tumor$PrePost &lt;- ifelse(arm3_tumor$Timepoint == 0, &quot;Baseline&quot;, &quot;Treated&quot;)

create_barplot(arm3_tumor,
               group_var = &quot;PrePost&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(arm3_tumor$Annotation))),
               group_levels = sort(unique(arm3_tumor$PrePost)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-30-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-30-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prop_test &lt;- sc_utils(arm3_tumor)
prop_test &lt;- permutation_test(
    prop_test, cluster_identity = &quot;Annotation&quot;,
    sample_1 = &quot;Baseline&quot;, sample_2 = &quot;Treated&quot;,
    sample_identity = &quot;PrePost&quot;)

permutation_plot(prop_test) +
  ggtitle(&quot;Baseline vs. treated&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-31-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-31-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="tumor-adjacent-cell-type-proportion-differences-between-timepoints-before-and-after-treatment"
class="section level2">
<h2>Tumor-adjacent cell type proportion differences between timepoints
(before and after treatment)</h2>
<pre class="r"><code>arm3_normal$PrePost &lt;- ifelse(arm3_normal$Timepoint == 0, &quot;Baseline&quot;, &quot;Treated&quot;)

create_barplot(arm3_normal,
               group_var = &quot;PrePost&quot;,
               plot_var = &quot;Annotation&quot;,
               plot_levels = sort((unique(arm3_normal$Annotation))),
               group_levels = sort(unique(arm3_normal$PrePost)),
               plot_colors = pipac_celltype_col,
               var_names =  c(&quot;Frequency (%)&quot;, &quot;&quot;),
               legend_title = &quot;Celltype&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-32-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-32-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>prop_test &lt;- sc_utils(arm3_normal)
prop_test &lt;- permutation_test(
    prop_test, cluster_identity = &quot;Annotation&quot;,
    sample_1 = &quot;Baseline&quot;, sample_2 = &quot;Treated&quot;,
    sample_identity = &quot;PrePost&quot;)

permutation_plot(prop_test) +
  ggtitle(&quot;Baseline vs. treated&quot;)</code></pre>
<p><img src="figure/arm3_comparative_analysis.Rmd/unnamed-chunk-33-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PIPAC_spatial/blob/3bbda1950b67a2205d9bebb3344530d4fbf8bcbd/docs/figure/arm3_comparative_analysis.Rmd/unnamed-chunk-33-1.png" target="_blank">3bbda19</a>
</td>
<td>
heinin
</td>
<td>
2025-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ComplexHeatmap_2.18.0       viridis_0.6.3              
 [3] viridisLite_0.4.2           circlize_0.4.15            
 [5] plyr_1.8.8                  RColorBrewer_1.1-3         
 [7] scProportionTest_0.0.0.9000 ggpmisc_0.5.2              
 [9] ggpp_0.5.2                  patchwork_1.1.2            
[11] googlesheets4_1.1.0         ggrepel_0.9.3              
[13] ggpubr_0.6.0                lubridate_1.9.2            
[15] forcats_1.0.0               stringr_1.5.0              
[17] dplyr_1.1.2                 purrr_1.0.2                
[19] readr_2.1.4                 tidyr_1.3.0                
[21] tibble_3.2.1                ggplot2_3.4.2              
[23] tidyverse_2.0.0             SeuratDisk_0.0.0.9021      
[25] Seurat_5.0.1                SeuratObject_5.0.2         
[27] sp_1.6-1                    arrow_12.0.0               
[29] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.20       splines_4.3.0          later_1.3.1           
  [4] cellranger_1.1.0       polyclip_1.10-4        fastDummies_1.7.3     
  [7] lifecycle_1.0.3        rstatix_0.7.2          doParallel_1.0.17     
 [10] rprojroot_2.0.3        globals_0.16.2         processx_3.8.1        
 [13] lattice_0.21-8         hdf5r_1.3.8            MASS_7.3-60           
 [16] backports_1.4.1        magrittr_2.0.3         plotly_4.10.2         
 [19] sass_0.4.6             rmarkdown_2.22         jquerylib_0.1.4       
 [22] yaml_2.3.7             httpuv_1.6.11          sctransform_0.4.1     
 [25] spam_2.9-1             spatstat.sparse_3.0-1  reticulate_1.29       
 [28] cowplot_1.1.1          pbapply_1.7-0          abind_1.4-5           
 [31] Rtsne_0.16             BiocGenerics_0.48.1    git2r_0.32.0          
 [34] S4Vectors_0.40.2       IRanges_2.36.0         irlba_2.3.5.1         
 [37] listenv_0.9.0          spatstat.utils_3.0-3   MatrixModels_0.5-1    
 [40] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.1-5 
 [43] fitdistrplus_1.1-11    parallelly_1.36.0      leiden_0.4.3          
 [46] codetools_0.2-19       shape_1.4.6            tidyselect_1.2.0      
 [49] farver_2.1.1           stats4_4.3.0           matrixStats_1.0.0     
 [52] spatstat.explore_3.2-1 googledrive_2.1.0      jsonlite_1.8.5        
 [55] GetoptLong_1.0.5       ellipsis_0.3.2         progressr_0.13.0      
 [58] iterators_1.0.14       ggridges_0.5.4         survival_3.5-5        
 [61] foreach_1.5.2          tools_4.3.0            ica_1.0-3             
 [64] Rcpp_1.0.10            glue_1.6.2             gridExtra_2.3         
 [67] xfun_0.39              withr_2.5.0            fastmap_1.1.1         
 [70] fansi_1.0.4            SparseM_1.81           callr_3.7.3           
 [73] digest_0.6.31          timechange_0.2.0       R6_2.5.1              
 [76] mime_0.12              colorspace_2.1-0       scattermore_1.2       
 [79] tensor_1.5             spatstat.data_3.0-1    utf8_1.2.3            
 [82] generics_0.1.3         data.table_1.14.8      httr_1.4.6            
 [85] htmlwidgets_1.6.2      whisker_0.4.1          uwot_0.1.14           
 [88] pkgconfig_2.0.3        gtable_0.3.3           lmtest_0.9-40         
 [91] htmltools_0.5.5        carData_3.0-5          dotCall64_1.0-2       
 [94] clue_0.3-64            scales_1.2.1           png_0.1-8             
 [97] knitr_1.43             rstudioapi_0.14        rjson_0.2.21          
[100] tzdb_0.4.0             reshape2_1.4.4         curl_5.0.1            
[103] nlme_3.1-162           GlobalOptions_0.1.2    cachem_1.0.8          
[106] zoo_1.8-12             KernSmooth_2.23-21     parallel_4.3.0        
[109] miniUI_0.1.1.1         pillar_1.9.0           vctrs_0.6.5           
[112] RANN_2.6.1             promises_1.2.0.1       car_3.1-2             
[115] xtable_1.8-4           cluster_2.1.4          evaluate_0.21         
[118] cli_3.6.1              compiler_4.3.0         rlang_1.1.1           
[121] crayon_1.5.2           future.apply_1.11.0    ggsignif_0.6.4        
[124] labeling_0.4.2         ps_1.7.5               getPass_0.2-4         
[127] fs_1.6.2               stringi_1.7.12         deldir_1.0-9          
[130] assertthat_0.2.1       munsell_0.5.0          lazyeval_0.2.2        
[133] spatstat.geom_3.2-1    quantreg_5.95          Matrix_1.6-5          
[136] RcppHNSW_0.5.0         hms_1.1.3              bit64_4.0.5           
[139] future_1.32.0          shiny_1.7.4            highr_0.10            
[142] ROCR_1.0-11            gargle_1.4.0           igraph_1.4.3          
[145] broom_1.0.4            bslib_0.4.2            bit_4.0.5             
[148] polynom_1.4-1         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
