<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="heinin" />

<meta name="date" content="2025-11-11" />

<title>Bulk-RNAseq analysis of CAF cell lines</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIPAC_spatial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Bulk-RNAseq analysis of CAF cell lines</h1>
<h4 class="author">heinin</h4>
<h4 class="date">2025-11-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-11-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>PIPAC_spatial/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240917code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240917)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240917code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240917)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/
</td>
<td style="text-align:left;">
.
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/code/PIPAC_colors_themes.R
</td>
<td style="text-align:left;">
code/PIPAC_colors_themes.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/code/plot_functions.R
</td>
<td style="text-align:left;">
code/plot_functions.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomheininPIPACspatialtree2221fdb25852e44537732db34a0abd5f08c4c9b0targetblank2221fdba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/heinin/PIPAC_spatial/tree/2221fdb25852e44537732db34a0abd5f08c4c9b0" target="_blank">2221fdb</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomheininPIPACspatialtree2221fdb25852e44537732db34a0abd5f08c4c9b0targetblank2221fdba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/heinin/PIPAC_spatial/tree/2221fdb25852e44537732db34a0abd5f08c4c9b0" target="_blank">2221fdb</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    analysis/figure/
    Ignored:    cell_immune_cluster_marker_annotations.tsv
    Ignored:    cell_nonimmune_cluster_marker_annotations.tsv
    Ignored:    celltype_markers.tsv
    Ignored:    code/.Rhistory
    Ignored:    code/Proximity_analysis/.DS_Store
    Ignored:    code/Proximity_analysis/data/
    Ignored:    code/Proximity_analysis/output/
    Ignored:    code/RSC_latest_EDM_2025-08-06/
    Ignored:    code/celltype_proximity_tumor.Rout
    Ignored:    immune_cluster_marker_annotations_2ndpass.tsv
    Ignored:    immune_prelim_annot_topmarkers.tsv
    Ignored:    nonimmune_cluster_marker_annotations_2ndpass.tsv
    Ignored:    nonimmune_prelim_annot_topmarkers.tsv

Untracked files:
    Untracked:  analysis/CAF_bulkRNAseq.Rmd
    Untracked:  analysis/finalize_annotations.Rmd
    Untracked:  analysis/immune_annotation_cell.Rmd
    Untracked:  analysis/nonimmune_annotation_cell.Rmd
    Untracked:  code/.RDataTmp
    Untracked:  code/.RDataTmp1
    Untracked:  code/celltype_proximity_09232025.R
    Untracked:  code/celltype_proximity_tumor.R
    Untracked:  code/denoist_chunk.R
    Untracked:  code/plot_informative_cluster_markers.R
    Untracked:  code/slurm.26717417.err
    Untracked:  code/slurm.26717417.out
    Untracked:  enrichment_posttreatment_scores_compiled.tsv
    Untracked:  enrichment_pretreatment_scores_compiled.tsv
    Untracked:  enrichment_tumor_scores_compiled.tsv
    Untracked:  output/enrichment_scores_compiled.tsv
    Untracked:  output/proximity_compiled.tsv
    Untracked:  tumor_posttreatment_proximity_compiled.tsv
    Untracked:  tumor_pretreatment_proximity_compiled.tsv
    Untracked:  tumor_proximity_compiled.tsv

Unstaged changes:
    Deleted:    Rplots.pdf
    Modified:   analysis/Xenium_preprocess_wholecell.Rmd
    Modified:   analysis/add_metadata.Rmd
    Modified:   analysis/annotation_cell.Rmd
    Modified:   analysis/arm3_comparative_analysis.Rmd
    Modified:   analysis/arm3_prepost_plots_DEGs.Rmd
    Modified:   analysis/compare_cellular_nuclear.Rmd
    Modified:   analysis/compare_cellular_nuclear_denoist_annot.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/niche_construction_ncells3k_nneighbors20.Rmd
    Modified:   analysis/pca_variance_decomp.Rmd
    Modified:   analysis/proximity.Rmd
    Modified:   analysis/splitting_samples.Rmd
    Modified:   code/PIPAC_colors_themes.R
    Modified:   code/anndata_to_seurat.R
    Modified:   code/denoist.R
    Modified:   code/pairwise_proximity.R
    Modified:   code/pairwise_proximity_chunk.R
    Modified:   code/parse_denoist_res.R
    Modified:   code/plot_functions.R
    Modified:   code/run_rscript.sh
    Modified:   code/seurat_to_anndata.R
    Modified:   code/update_metadata.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="packages-and-environment-variables" class="section level3">
<h3>Packages and environment variables</h3>
<pre class="r"><code>#install.packages(&quot;ggfun&quot;, lib=&quot;/home/hnatri/R/rstudio-4.3.0-4-with_modules.sif/libs&quot;)
#packageurl &lt;- &quot;https://cran.r-project.org/src/contrib/ggfun_0.2.0.tar.gz&quot;
#install.packages(packageurl, repos=NULL, type=&quot;source&quot;)
#remotes::install_github(&quot;YuLab-SMU/aplot&quot;, lib=&quot;/home/hnatri/R/rstudio-4.3.0-4-with_modules.sif/libs&quot;)
#BiocManager::install(&quot;clusterProfiler&quot;, lib=&quot;/home/hnatri/R/rstudio-4.3.0-4-with_modules.sif/libs&quot;)

suppressPackageStartupMessages({
  library(workflowr)
  library(tidyverse)
  library(edgeR)
  library(limma)
  library(plyr)
  library(EnhancedVolcano)
  library(patchwork)
  library(contsurvplot)
  library(survival)
  library(survminer)
  library(googlesheets4)
  library(clusterProfiler)
  library(org.Hs.eg.db)})</code></pre>
</div>
<div id="environment-variables-and-helper-functions"
class="section level3">
<h3>Environment variables and helper functions</h3>
<pre class="r"><code>setwd(&quot;/home/hnatri/PIPAC_spatial/&quot;)
set.seed(9999)
options(scipen = 99999)
options(ggrepel.max.overlaps = Inf)

source(&quot;/home/hnatri/PIPAC_spatial/code/PIPAC_colors_themes.R&quot;)
source(&quot;/home/hnatri/PIPAC_spatial/code/plot_functions.R&quot;)</code></pre>
</div>
<div id="marker-information-and-metadata" class="section level3">
<h3>Marker information and metadata</h3>
<pre class="r"><code># Cell type markers by group
gs4_deauth()
markers_by_group  &lt;- gs4_get(&quot;https://docs.google.com/spreadsheets/d/1w0wrL-5KwNEWi_VpmriN7YmfwpBPlEYX0MKlwI2ZQu8/edit?usp=sharing&quot;)
markers_by_group &lt;- read_sheet(markers_by_group, sheet = &quot;Markers by group&quot;)</code></pre>
</div>
<div id="import-data" class="section level3">
<h3>Import data</h3>
<pre class="r"><code># Count data with genes in rows and samples in columns
human_counts &lt;- read.table(&quot;/tgen_labs/banovich/Raoof_Appendiceal_Bulk/CAF_celllines/CAF_human_gene_count.tsv&quot;,
                            header = T, sep = &quot;\t&quot;)
mouse_counts &lt;- read.table(&quot;/tgen_labs/banovich/Raoof_Appendiceal_Bulk/CAF_celllines/CAF_mouse_gene_count.tsv&quot;,
                           header = T, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="human" class="section level2">
<h2>Human</h2>
<div id="preprocessing" class="section level3">
<h3>Preprocessing</h3>
<pre class="r"><code>all_counts_df &lt;- human_counts
gene_info &lt;- all_counts_df[,grep(&quot;gene&quot;, colnames(all_counts_df))]
rownames(all_counts_df) &lt;- all_counts_df$gene_id
all_counts_df &lt;- all_counts_df[,grep(&quot;COL&quot;, colnames(all_counts_df))]

# Excluding ribosomal genes
ribo &lt;- gene_info[grep(&quot;^RP&quot;, gene_info$gene_name),]
all_counts_df &lt;- all_counts_df[!(row.names(all_counts_df) %in% ribo$gene_id), ]

# Arranging gene info to match the count matrix
gene_info &lt;- gene_info %&gt;%
  filter(gene_id %in% rownames(all_counts_df)) %&gt;%
  dplyr::arrange(factor(gene_id, levels = rownames(all_counts_df)))

identical(gene_info$gene_id, rownames(all_counts_df))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code># Calculating gene length
#gene_info$length &lt;- abs(gene_info$stop - gene_info$start)

# Create DGEList object
d0 &lt;- DGEList(all_counts_df, genes = gene_info)

# Calculate normalization factors
d0 &lt;- calcNormFactors(d0)
d0</code></pre>
<pre><code>An object of class &quot;DGEList&quot;
$counts
                 COL4_V COL5_V COL7_V COL4_MMC COL5_MMC COL7_MMC
ENSG00000115414 1007243 875459 942098  1246953  1010717   941199
ENSG00000108821  713824 978378 892653   843417   853001   852083
ENSG00000164692  454770 702554 655268   560478   728775   682148
ENSG00000113140  287728 297595 286634   357609   340819   315493
ENSG00000133110  298287 340923 256292   274959   348530   246149
23914 more rows ...

$samples
         group lib.size norm.factors
COL4_V       1 19959680    0.9453843
COL5_V       1 19966684    0.9826067
COL7_V       1 18390210    0.9819295
COL4_MMC     1 23452779    0.9815971
COL5_MMC     1 21117282    1.0546043
COL7_MMC     1 19024618    1.0590314

$genes
                        gene_id gene_name gene_chr gene_start  gene_end
ENSG00000115414 ENSG00000115414       FN1        2  215360440 215436172
ENSG00000108821 ENSG00000108821    COL1A1       17   50183289  50201632
ENSG00000164692 ENSG00000164692    COL1A2        7   94394561  94431232
ENSG00000113140 ENSG00000113140     SPARC        5  151661096 151687165
ENSG00000133110 ENSG00000133110     POSTN       13   37562583  37598844
                gene_strand gene_length   gene_biotype
ENSG00000115414           -       17315 protein_coding
ENSG00000108821           -        9819 protein_coding
ENSG00000164692           +       11156 protein_coding
ENSG00000113140           -        4831 protein_coding
ENSG00000133110           -        5464 protein_coding
                                                                             gene_description
ENSG00000115414                              fibronectin 1 [Source:HGNC Symbol;Acc:HGNC:3778]
ENSG00000108821              collagen type I alpha 1 chain [Source:HGNC Symbol;Acc:HGNC:2197]
ENSG00000164692              collagen type I alpha 2 chain [Source:HGNC Symbol;Acc:HGNC:2198]
ENSG00000113140 secreted protein acidic and cysteine rich [Source:HGNC Symbol;Acc:HGNC:11219]
ENSG00000133110                                 periostin [Source:HGNC Symbol;Acc:HGNC:16953]
23914 more rows ...</code></pre>
<pre class="r"><code># Filter low-expressed genes
cutoff &lt;- 1
drop &lt;- which(apply(rpkm(d0, gene.length = &quot;gene_length&quot;), 1, max) &lt; cutoff)
d &lt;- d0[-drop,] 
dim(d) # number of genes left</code></pre>
<pre><code>[1] 7985    6</code></pre>
<pre class="r"><code># Matching sample names with patient metadata
#appendiceal_metadata &lt;- merge(appendiceal_metadata, sample_key, by = &quot;Patient.ID&quot;)
#appendiceal_metadata &lt;- appendiceal_metadata %&gt;%
#  filter(X2025.HonestBroker.specimen.number %in% colnames(all_counts_df)) %&gt;%
#  dplyr::arrange(match(X2025.HonestBroker.specimen.number, colnames(all_counts_df)))

# Dimensionality reduction colored with various metadata
group_info &lt;- sapply(strsplit(colnames(all_counts_df), &quot;_&quot;), `[`, 2)
sample_info &lt;-  sapply(strsplit(colnames(all_counts_df), &quot;_&quot;), `[`, 1)

# Set colors and shapes
cols &lt;- c(&quot;deeppink3&quot;, &quot;cyan4&quot;, &quot;yellow3&quot;)
#shps &lt;- c(15, 16, 17, 18, 15, 16, 17, 18)
shps &lt;- c(15, 15, 15, 16, 16, 16)
names(cols) &lt;- unique(sample_info)
names(shps) &lt;- group_info

plotMDS(d, top = 500, col = cols, pch = shps, cex = 1, labels = colnames(all_counts_df))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-5-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#legend(&#39;bottomright&#39;, col=cols, legend=unique(sample_info), pch = 16, cex = 0.7)
#legend(&#39;topright&#39;, legend=unique(group_info), pch = shps, cex = 0.7)</code></pre>
<pre class="r"><code># Total number of reads per sample
d$samples %&gt;% 
  rownames_to_column(var = &quot;sample&quot;) %&gt;%
  ggplot(aes(x = reorder(sample, -lib.size), y = lib.size)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  theme(plot.margin = unit(c(0,0,0,1), &quot;cm&quot;)) +
  xlab(&quot;Sample&quot;) + 
  ylab(&quot;Library size&quot;)</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-6-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="de-analysis" class="section level3">
<h3>DE analysis</h3>
<pre class="r"><code>#group_info &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 1)
#sample_info &lt;- c(&quot;COL4&quot;, &quot;COL5&quot;, &quot;COL7&quot;, &quot;COL4&quot;, &quot;COL5&quot;, &quot;COL7&quot;)

# Voom transformation and calculation of variance weights.
# Specify the model to be fitted. We do this before using voom since voom uses
# variances of the model residuals (observed - fitted).
# Edit to use the relevant metadata column.
groups &lt;- data.frame(sample_info = sample_info, group_info = group_info)

#group_info2 &lt;- c(&quot;V&quot;, &quot;V&quot;, &quot;MMC&quot;, &quot;MMC&quot;, &quot;V&quot;, &quot;V&quot;, &quot;MMC&quot;, &quot;MMC&quot;)

mm &lt;- model.matrix(~0 + sample_info + group_info)
# Renaming columns
#colnames(mm) &lt;- c(&quot;MMC&quot;, &quot;V&quot;)

# Filtering based on the design matrix
# Defaults: min.count = 10, min.total.count = 15, large.n = 10, min.prop = 0.7
keep &lt;- filterByExpr(d, mm)
d1 &lt;- d[keep,]

# Replotting MDS and library sizes
plotMDS(d1, top = 500, col = cols, pch = shps)
legend(&#39;bottomright&#39;, col=cols, legend=unique(sample_info), pch = 16, cex = 0.7)
legend(&#39;topright&#39;, legend=unique(group_info), pch = shps, cex = 0.7)</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-7-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Total number of reads per sample
d1$samples %&gt;% 
  rownames_to_column(var = &quot;sample&quot;) %&gt;%
  ggplot(aes(x = reorder(sample, -lib.size), y = lib.size)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  theme(plot.margin = unit(c(0,0,0,1), &quot;cm&quot;)) +
  xlab(&quot;Sample&quot;) + 
  ylab(&quot;Library size&quot;)</code></pre>
<pre class="r"><code># The above specifies a model where each coefficient corresponds to a group mean
# Voom
y1 &lt;- voom(d1, mm, plot = T)</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-9-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fitting linear models in limma
# lmFit fits a linear model using weighted least squares for each gene:
fit1 &lt;- lmFit(y1, mm)
head(coef(fit1))</code></pre>
<pre><code>                sample_infoCOL4 sample_infoCOL5 sample_infoCOL7  group_infoV
ENSG00000115414        15.69563        15.43894        15.57209  0.037824285
ENSG00000108821        15.07516        15.30641        15.37147  0.218346846
ENSG00000164692        14.52805        15.02653        15.06073  0.072799456
ENSG00000113140        13.91322        13.89857        13.94779 -0.006725222
ENSG00000133110        13.61773        13.88088        13.55633  0.256695834
ENSG00000137801        13.87324        13.77613        12.91705  0.279704991</code></pre>
<pre class="r"><code># Specify comparisons
contr1 &lt;- makeContrasts(colnames(coef(fit1)), levels = colnames(coef(fit1)))
contr1</code></pre>
<pre><code>                 Contrasts
Levels            sample_infoCOL4
  sample_infoCOL4               0
  sample_infoCOL5               0
  sample_infoCOL7               0
  group_infoV                   1</code></pre>
<pre class="r"><code>tmp &lt;- contrasts.fit(fit1, contr1)
tmp &lt;- eBayes(tmp)

top.table &lt;- topTable(tmp, sort.by = &quot;P&quot;, n = Inf)
top.table$gene_name &lt;- mapvalues(x = rownames(top.table),
                                 from = gene_info$gene_id,
                                 to = gene_info$gene_name)

# Saving all results
write.table(top.table, &quot;/home/hnatri/Raoof_Appendiceal_Bulk/human_MMC_V_res.tsv&quot;,
            sep = &quot;\t&quot;, quote = F, row.names = T)

# Plotting
top_sig &lt;- top.table %&gt;%
  filter(adj.P.Val &lt; 0.05 &amp; logFC &gt; 1)
  #dplyr::select(gene_name)

p0 &lt;- EnhancedVolcano(top.table,
                      lab = top.table$gene_name,
                      x = &#39;logFC&#39;,
                      y = &#39;adj.P.Val&#39;,
                      title = &quot;&quot;,
                      subtitle = &quot;&quot;,
                      pCutoff = 0.05,
                      FCcutoff = 1,
                      ylim = c(0, max(-log10(top.table[[&quot;adj.P.Val&quot;]]), na.rm = TRUE) + 0.5),
                      caption = &quot;&quot;,
                      titleLabSize = 12,
                      subtitleLabSize = 2,
                      legendLabSize = 10,
                      axisLabSize = 10,
                      legendIconSize = 3,
                      labSize = 4)

p0</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-9-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="fpkm-for-each-gene" class="section level3">
<h3>FPKM for each gene</h3>
<pre class="r"><code># Calculating FPKM for each gene
metadata_expression &lt;- rpkm(d0, gene.length = &quot;gene_length&quot;) %&gt;% t() %&gt;% as.data.frame() %&gt;%
  rownames_to_column(var = &quot;Sample_Treatment&quot;)

metadata_expression$Sample &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 1)
metadata_expression$Treatment &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 2)

gene_info %&gt;% filter(gene_name == &quot;Tenm4&quot;)

# Expression by group
p1 &lt;- metadata_expression %&gt;%
  ggplot(aes(x = Treatment, y = ENSMUSG00000048078, fill = Treatment)) +
  geom_violin(alpha = 0.5) +
  geom_point(position = position_jitter(seed = 1, width = 0.2)) +
  ylab(&quot;Tenm4 expression (FPKM)&quot;) +
  xlab(&quot;Treatment&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

p1</code></pre>
</div>
<div id="plotting-marker-genes" class="section level3">
<h3>Plotting marker genes</h3>
<pre class="r"><code># Mouse to human gene names
mouse_human_genes &lt;- read.table(&quot;http://www.informatics.jax.org/downloads/reports/HOM_MouseHumanSequence.rpt&quot;,
                                sep=&quot;\t&quot;, header = T)

mouse &lt;- split.data.frame(mouse_human_genes,mouse_human_genes$Common.Organism.Name)[[2]]
human &lt;- split.data.frame(mouse_human_genes,mouse_human_genes$Common.Organism.Name)[[1]]
mouse &lt;- mouse[,c(1,4)]
human &lt;- human[,c(1,4)]
mh_data &lt;- merge.data.frame(mouse, human, by = &quot;DB.Class.Key&quot;,all.y = TRUE)

# Markers to plot
mycafs &lt;- c(&quot;TGFB1&quot;, &quot;ACTA2&quot;, &quot;TAGLN&quot;, &quot;HOPX&quot;, &quot;MYL9&quot;)
icafs &lt;- c(&quot;IL6&quot;, &quot;CXCL12&quot;, &quot;CXCL13&quot;, &quot;CXCL14&quot;)
abcafs &lt;- c(&quot;CD74&quot;, &quot;HLA-DPB1&quot;, &quot;HLA-DPB2&quot;, &quot;HLA-DQA1&quot;, &quot;HLA-DRA&quot;, &quot;HLA-DRB1&quot;)
ficafs &lt;- c(&quot;SFRP2&quot;, &quot;SFRP4&quot;, &quot;COMP&quot;, &quot;CXCL14&quot;, &quot;PLA2G2A&quot;, &quot;ASPN&quot;)

all_caf_markers &lt;- c(mycafs, icafs, abcafs, ficafs)

top_sig %&gt;% filter(gene_name %in% all_caf_markers)</code></pre>
<pre><code> [1] gene_id          gene_name        gene_chr         gene_start      
 [5] gene_end         gene_strand      gene_length      gene_biotype    
 [9] gene_description logFC            AveExpr          t               
[13] P.Value          adj.P.Val        B               
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code># FPKM, no filtering
# Create DGEList object
d0 &lt;- DGEList(all_counts_df, genes = gene_info)

# Calculate normalization factors
d0 &lt;- calcNormFactors(d0)
d0</code></pre>
<pre><code>An object of class &quot;DGEList&quot;
$counts
                 COL4_V COL5_V COL7_V COL4_MMC COL5_MMC COL7_MMC
ENSG00000115414 1007243 875459 942098  1246953  1010717   941199
ENSG00000108821  713824 978378 892653   843417   853001   852083
ENSG00000164692  454770 702554 655268   560478   728775   682148
ENSG00000113140  287728 297595 286634   357609   340819   315493
ENSG00000133110  298287 340923 256292   274959   348530   246149
23914 more rows ...

$samples
         group lib.size norm.factors
COL4_V       1 19959680    0.9453843
COL5_V       1 19966684    0.9826067
COL7_V       1 18390210    0.9819295
COL4_MMC     1 23452779    0.9815971
COL5_MMC     1 21117282    1.0546043
COL7_MMC     1 19024618    1.0590314

$genes
                        gene_id gene_name gene_chr gene_start  gene_end
ENSG00000115414 ENSG00000115414       FN1        2  215360440 215436172
ENSG00000108821 ENSG00000108821    COL1A1       17   50183289  50201632
ENSG00000164692 ENSG00000164692    COL1A2        7   94394561  94431232
ENSG00000113140 ENSG00000113140     SPARC        5  151661096 151687165
ENSG00000133110 ENSG00000133110     POSTN       13   37562583  37598844
                gene_strand gene_length   gene_biotype
ENSG00000115414           -       17315 protein_coding
ENSG00000108821           -        9819 protein_coding
ENSG00000164692           +       11156 protein_coding
ENSG00000113140           -        4831 protein_coding
ENSG00000133110           -        5464 protein_coding
                                                                             gene_description
ENSG00000115414                              fibronectin 1 [Source:HGNC Symbol;Acc:HGNC:3778]
ENSG00000108821              collagen type I alpha 1 chain [Source:HGNC Symbol;Acc:HGNC:2197]
ENSG00000164692              collagen type I alpha 2 chain [Source:HGNC Symbol;Acc:HGNC:2198]
ENSG00000113140 secreted protein acidic and cysteine rich [Source:HGNC Symbol;Acc:HGNC:11219]
ENSG00000133110                                 periostin [Source:HGNC Symbol;Acc:HGNC:16953]
23914 more rows ...</code></pre>
<pre class="r"><code># Combining expression and metadata
metadata_expression &lt;- rpkm(d0, gene.length = &quot;gene_length&quot;) %&gt;% t() %&gt;% as.data.frame() %&gt;%
  rownames_to_column(var = &quot;Sample_Treatment&quot;)

metadata_expression$Sample &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 1)
metadata_expression$Treatment &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 2)

long_df &lt;- metadata_expression %&gt;%
  pivot_longer(cols = grep(&quot;ENS&quot;, colnames(metadata_expression), value = T),
               names_to = &quot;ensembl&quot;, values_to = &quot;exp&quot;)

# Adding gene names
long_df$gene &lt;- mapvalues(x = long_df$ensembl,
                          from = gene_info$gene_id,
                          to = gene_info$gene_name)

long_df$Hh_gene &lt;- mapvalues(x = long_df$gene,
                             from = mh_data$Symbol.x,
                             to = mh_data$Symbol.y)

long_df$Treatment_simple &lt;- ifelse(long_df$Treatment %in% c(&quot;V1&quot;, &quot;V2&quot;), &quot;V&quot;, &quot;MMC&quot;)

# Plotting
long_df %&gt;% filter(Hh_gene %in% icafs) %&gt;%
  ggplot(aes(x = Treatment_simple, y = exp, color = Treatment_simple, label = Sample)) +
  geom_violin(alpha = 0.5) +
  geom_point() +
  geom_text(hjust=0, vjust=0) +
  #geom_line(aes(group = Sample), color = &quot;gray20&quot;) +
  facet_wrap(~Hh_gene, scales = &quot;free&quot;) +
  ylab(&quot;iCAF expression (FPKM)&quot;) +
  xlab(&quot;&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-11-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>long_df %&gt;% filter(Hh_gene %in% mycafs) %&gt;%
  ggplot(aes(x = Treatment_simple, y = exp, color = Treatment_simple, label = Sample)) +
  geom_violin(alpha = 0.5) +
  geom_point() +
  geom_text(hjust=0, vjust=0) +
  #geom_line(aes(group = Sample), color = &quot;gray20&quot;) +
  facet_wrap(~Hh_gene, scales = &quot;free&quot;) +
  ylab(&quot;myCAF expression (FPKM)&quot;) +
  xlab(&quot;&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-11-2.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>long_df %&gt;% filter(Hh_gene %in% abcafs) %&gt;%
  ggplot(aes(x = Treatment_simple, y = exp, color = Treatment_simple, label = Sample)) +
  geom_violin(alpha = 0.5) +
  geom_point() +
  geom_text(hjust=0, vjust=0) +
  #geom_line(aes(group = Sample), color = &quot;gray20&quot;) +
  facet_wrap(~Hh_gene, scales = &quot;free&quot;) +
  ylab(&quot;abCAF expression (FPKM)&quot;) +
  xlab(&quot;&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-11-3.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>long_df %&gt;% filter(Hh_gene %in% ficafs) %&gt;%
  ggplot(aes(x = Treatment_simple, y = exp, color = Treatment_simple, label = Sample)) +
  geom_violin(alpha = 0.5) +
  geom_point() +
  geom_text(hjust=0, vjust=0) +
  #geom_line(aes(group = Sample), color = &quot;gray20&quot;) +
  facet_wrap(~Hh_gene, scales = &quot;free&quot;) +
  ylab(&quot;fiCAF expression (FPKM)&quot;) +
  xlab(&quot;&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-11-4.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="mouse" class="section level2">
<h2>Mouse</h2>
<div id="preprocessing-1" class="section level3">
<h3>Preprocessing</h3>
<pre class="r"><code># Excluding ribosomal genes
all_counts_df &lt;- mouse_counts
gene_info &lt;- all_counts_df[,grep(&quot;gene&quot;, colnames(all_counts_df))]
rownames(all_counts_df) &lt;- all_counts_df$gene_id
all_counts_df &lt;- all_counts_df[,grep(&quot;CAF&quot;, colnames(all_counts_df))]

ribo &lt;- gene_info[grep(&quot;^Rp&quot;, gene_info$gene_name),]
all_counts_df &lt;- all_counts_df[!(rownames(all_counts_df) %in% ribo$gene_id), ]

# Arranging gene info to match the count matrix
gene_info &lt;- gene_info %&gt;%
  filter(gene_id %in% rownames(all_counts_df)) %&gt;%
  dplyr::arrange(factor(gene_id, levels = rownames(all_counts_df)))

identical(gene_info$gene_id, rownames(all_counts_df))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code># Calculating gene length
#gene_info$length &lt;- abs(gene_info$stop - gene_info$start)

# Create DGEList object
d0 &lt;- DGEList(all_counts_df, genes = gene_info)

# Calculate normalization factors
d0 &lt;- calcNormFactors(d0)
d0</code></pre>
<pre><code>An object of class &quot;DGEList&quot;
$counts
                   CAF1_V1 CAF1_V2 CAF1_MMC1 CAF1_MMC2 CAF2_V1 CAF2_V2
ENSMUSG00000026193  313278  368553    272989    230151  153189  148543
ENSMUSG00000040152  372316  426808    304702    260705   65276   63117
ENSMUSG00000001506  216965  255796    160159    143006  126065  128695
ENSMUSG00000029661  173567  197616    131673    118101  148318  145185
ENSMUSG00000037742  109253  122688    121259     99121  123691  131161
                   CAF2_MMC1 CAF2_MMC2
ENSMUSG00000026193    136134    143371
ENSMUSG00000040152     52535     58877
ENSMUSG00000001506    116504    107571
ENSMUSG00000029661    118020    114836
ENSMUSG00000037742    107255    119192
28741 more rows ...

$samples
          group lib.size norm.factors
CAF1_V1       1 12175504    0.8866511
CAF1_V2       1 13818374    0.8720662
CAF1_MMC1     1 12434515    0.9373871
CAF1_MMC2     1 10427362    0.9415915
CAF2_V1       1 10341126    1.0687947
CAF2_V2       1 10380767    1.0713266
CAF2_MMC1     1 10245181    1.1215404
CAF2_MMC2     1 10814496    1.1409987

$genes
                              gene_id gene_name gene_chr gene_start  gene_end
ENSMUSG00000026193 ENSMUSG00000026193       Fn1        1   71624679  71692359
ENSMUSG00000040152 ENSMUSG00000040152     Thbs1        2  117942357 117957614
ENSMUSG00000001506 ENSMUSG00000001506    Col1a1       11   94827050  94843868
ENSMUSG00000029661 ENSMUSG00000029661    Col1a2        6    4504814   4541544
ENSMUSG00000037742 ENSMUSG00000037742    Eef1a1        9   78385731  78396433
                   gene_strand gene_length   gene_biotype
ENSMUSG00000026193           -       11685 protein_coding
ENSMUSG00000040152           +        6073 protein_coding
ENSMUSG00000001506           +        6563 protein_coding
ENSMUSG00000029661           +       12426 protein_coding
ENSMUSG00000037742           -        2493 protein_coding
                                                                                         gene_description
ENSMUSG00000026193                                        fibronectin 1 [Source:MGI Symbol;Acc:MGI:95566]
ENSMUSG00000040152                                     thrombospondin 1 [Source:MGI Symbol;Acc:MGI:98737]
ENSMUSG00000001506                            collagen, type I, alpha 1 [Source:MGI Symbol;Acc:MGI:88467]
ENSMUSG00000029661                            collagen, type I, alpha 2 [Source:MGI Symbol;Acc:MGI:88468]
ENSMUSG00000037742 eukaryotic translation elongation factor 1 alpha 1 [Source:MGI Symbol;Acc:MGI:1096881]
28741 more rows ...</code></pre>
<pre class="r"><code># Filter low-expressed genes
cutoff &lt;- 1
drop &lt;- which(apply(rpkm(d0, gene.length = &quot;gene_length&quot;), 1, max) &lt; cutoff)
d &lt;- d0[-drop,] 
dim(d) # number of genes left</code></pre>
<pre><code>[1] 5787    8</code></pre>
<pre class="r"><code># Matching sample names with patient metadata
#appendiceal_metadata &lt;- merge(appendiceal_metadata, sample_key, by = &quot;Patient.ID&quot;)
#appendiceal_metadata &lt;- appendiceal_metadata %&gt;%
#  filter(X2025.HonestBroker.specimen.number %in% colnames(all_counts_df)) %&gt;%
#  dplyr::arrange(match(X2025.HonestBroker.specimen.number, colnames(all_counts_df)))

# Dimensionality reduction colored with various metadata
group_info &lt;- sapply(strsplit(colnames(all_counts_df), &quot;_&quot;), `[`, 2)
sample_info &lt;-  sapply(strsplit(colnames(all_counts_df), &quot;_&quot;), `[`, 1)

# Set colors and shapes
cols &lt;- c(&quot;deeppink3&quot;, &quot;cyan4&quot;)
shps &lt;- c(15, 16, 17, 18, 15, 16, 17, 18)
#shps &lt;- c(15, 15, 15, 16, 16, 16)
names(cols) &lt;- unique(sample_info)
names(shps) &lt;- unique(group_info)

plotMDS(d, top = 500, pch = shps, col = cols, cex = 1)</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-12-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#legend(&#39;bottomright&#39;, col=cols, legend=unique(sample_info), pch = 16, cex = 0.7)
#legend(&#39;topright&#39;, legend=unique(group_info), pch = shps, cex = 0.7)</code></pre>
<pre class="r"><code># Total number of reads per sample
d$samples %&gt;% 
  rownames_to_column(var = &quot;sample&quot;) %&gt;%
  ggplot(aes(x = reorder(sample, -lib.size), y = lib.size)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  theme(plot.margin = unit(c(0,0,0,1), &quot;cm&quot;)) +
  xlab(&quot;Sample&quot;) + 
  ylab(&quot;Library size&quot;)</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-13-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="de-analysis-1" class="section level3">
<h3>DE analysis</h3>
<pre class="r"><code>#group_info &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 1)
#sample_info &lt;- c(&quot;COL4&quot;, &quot;COL5&quot;, &quot;COL7&quot;, &quot;COL4&quot;, &quot;COL5&quot;, &quot;COL7&quot;)

# Voom transformation and calculation of variance weights.
# Specify the model to be fitted. We do this before using voom since voom uses
# variances of the model residuals (observed - fitted).
# Edit to use the relevant metadata column.
groups &lt;- data.frame(sample_info = sample_info, group_info = group_info)

#group_info2 &lt;- c(&quot;V&quot;, &quot;V&quot;, &quot;MMC&quot;, &quot;MMC&quot;, &quot;V&quot;, &quot;V&quot;, &quot;MMC&quot;, &quot;MMC&quot;)

mm &lt;- model.matrix(~0 + sample_info + group_info)
# Renaming columns
#colnames(mm) &lt;- c(&quot;MMC&quot;, &quot;V&quot;)

# Filtering based on the design matrix
# Defaults: min.count = 10, min.total.count = 15, large.n = 10, min.prop = 0.7
keep &lt;- filterByExpr(d, mm)
d1 &lt;- d[keep,]

# Replotting MDS and library sizes
plotMDS(d1, top = 500, col = cols, pch = shps)
legend(&#39;bottomright&#39;, col=cols, legend=unique(sample_info), pch = 16, cex = 0.7)
legend(&#39;topright&#39;, legend=unique(group_info), pch = shps, cex = 0.7)</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-14-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Total number of reads per sample
d1$samples %&gt;% 
  rownames_to_column(var = &quot;sample&quot;) %&gt;%
  ggplot(aes(x = reorder(sample, -lib.size), y = lib.size)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  theme(plot.margin = unit(c(0,0,0,1), &quot;cm&quot;)) +
  xlab(&quot;Sample&quot;) + 
  ylab(&quot;Library size&quot;)</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-15-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code># The above specifies a model where each coefficient corresponds to a group mean
# Voom
y1 &lt;- voom(d1, mm, plot = T)</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-16-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fitting linear models in limma
# lmFit fits a linear model using weighted least squares for each gene:
fit1 &lt;- lmFit(y1, mm)
head(coef(fit1))</code></pre>
<pre><code>                   sample_infoCAF1 sample_infoCAF2 group_infoMMC2 group_infoV1
ENSMUSG00000026193        14.55596        13.49168   -0.013386837    0.2678738
ENSMUSG00000040152        14.68862        12.14376    0.041897683    0.3847080
ENSMUSG00000001506        13.88551        13.16751   -0.066494772    0.3597286
ENSMUSG00000029661        13.53204        13.25772   -0.025779061    0.4475282
ENSMUSG00000037742        13.24935        13.28372    0.002962901    0.1110102
ENSMUSG00000119584        13.82740        12.97272   -0.383403805   -0.1085237
                   group_infoV2
ENSMUSG00000026193    0.2791414
ENSMUSG00000040152    0.3755523
ENSMUSG00000001506    0.4096242
ENSMUSG00000029661    0.4419613
ENSMUSG00000037742    0.1531156
ENSMUSG00000119584   -0.2013886</code></pre>
<pre class="r"><code># Specify comparisons
contr1 &lt;- makeContrasts(colnames(coef(fit1)), levels = colnames(coef(fit1)))
contr1</code></pre>
<pre><code>                 Contrasts
Levels            sample_infoCAF1
  sample_infoCAF1               0
  sample_infoCAF2               0
  group_infoMMC2                0
  group_infoV1                  0
  group_infoV2                  1</code></pre>
<pre class="r"><code>tmp &lt;- contrasts.fit(fit1, contr1)
tmp &lt;- eBayes(tmp)

top.table.mouse &lt;- topTable(tmp, sort.by = &quot;P&quot;, n = Inf)
top.table.mouse$gene_name &lt;- mapvalues(x = rownames(top.table.mouse),
                                 from = gene_info$gene_id,
                                 to = gene_info$gene_name)

# Saving all results
write.table(top.table.mouse, &quot;/home/hnatri/Raoof_Appendiceal_Bulk/mouse_MMC_V_res.tsv&quot;,
            sep = &quot;\t&quot;, quote = F, row.names = T)

# Plotting
top_sig &lt;- top.table.mouse %&gt;%
  filter(adj.P.Val &lt; 0.05 &amp; logFC &gt; 1)
  #dplyr::select(gene_name)

p0 &lt;- EnhancedVolcano(top.table.mouse,
                      lab = top.table.mouse$gene_name,
                      x = &#39;logFC&#39;,
                      y = &#39;adj.P.Val&#39;,
                      title = &quot;&quot;,
                      subtitle = &quot;&quot;,
                      pCutoff = 0.05,
                      FCcutoff = 1,
                      ylim = c(0, max(-log10(top.table.mouse[[&quot;adj.P.Val&quot;]]), na.rm = TRUE) + 0.5),
                      caption = &quot;&quot;,
                      titleLabSize = 12,
                      subtitleLabSize = 2,
                      legendLabSize = 10,
                      axisLabSize = 10,
                      legendIconSize = 3,
                      labSize = 4)

p0</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-16-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="fpkm-for-each-gene-1" class="section level3">
<h3>FPKM for each gene</h3>
<pre class="r"><code># Calculating FPKM for each gene
metadata_expression &lt;- rpkm(d0, gene.length = &quot;gene_length&quot;) %&gt;% t() %&gt;% as.data.frame() %&gt;%
  rownames_to_column(var = &quot;Sample_Treatment&quot;)

metadata_expression$Sample &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 1)
metadata_expression$Treatment &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 2)

gene_info %&gt;% filter(gene_name == &quot;Tenm4&quot;)

# Expression by group
p1 &lt;- metadata_expression %&gt;%
  ggplot(aes(x = Treatment, y = ENSMUSG00000048078, fill = Treatment)) +
  geom_violin(alpha = 0.5) +
  geom_point(position = position_jitter(seed = 1, width = 0.2)) +
  ylab(&quot;Tenm4 expression (FPKM)&quot;) +
  xlab(&quot;Treatment&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

p1</code></pre>
</div>
<div id="plotting-marker-genes-1" class="section level3">
<h3>Plotting marker genes</h3>
<pre class="r"><code># Mouse to human gene names
# Converting human gene names to mouse
mouse_human_genes &lt;- read.table(&quot;http://www.informatics.jax.org/downloads/reports/HOM_MouseHumanSequence.rpt&quot;,
                                sep=&quot;\t&quot;, header = T)

mouse &lt;- split.data.frame(mouse_human_genes,mouse_human_genes$Common.Organism.Name)[[2]]
human &lt;- split.data.frame(mouse_human_genes,mouse_human_genes$Common.Organism.Name)[[1]]
mouse &lt;- mouse[,c(1,4)]
human &lt;- human[,c(1,4)]
mh_data &lt;- merge.data.frame(mouse, human, by = &quot;DB.Class.Key&quot;,all.y = TRUE)

# Markers to plot
mycafs &lt;- c(&quot;TGFB1&quot;, &quot;ACTA2&quot;, &quot;TAGLN&quot;, &quot;HOPX&quot;, &quot;MYL9&quot;)
icafs &lt;- c(&quot;IL6&quot;, &quot;CXCL12&quot;, &quot;CXCL13&quot;, &quot;CXCL14&quot;)
abcafs &lt;- c(&quot;CD74&quot;, &quot;HLA-DPB1&quot;, &quot;HLA-DPB2&quot;, &quot;HLA-DQA1&quot;, &quot;HLA-DRA&quot;, &quot;HLA-DRB1&quot;)
ficafs &lt;- c(&quot;SFRP2&quot;, &quot;SFRP4&quot;, &quot;COMP&quot;, &quot;CXCL14&quot;, &quot;PLA2G2A&quot;, &quot;ASPN&quot;)

all_caf_markers &lt;- c(mycafs, icafs, abcafs, ficafs)

top_sig %&gt;% filter(gene_name %in% all_caf_markers)</code></pre>
<pre><code> [1] gene_id          gene_name        gene_chr         gene_start      
 [5] gene_end         gene_strand      gene_length      gene_biotype    
 [9] gene_description logFC            AveExpr          t               
[13] P.Value          adj.P.Val        B               
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code># FPKM, no filtering
# Create DGEList object
d0 &lt;- DGEList(all_counts_df, genes = gene_info)

# Calculate normalization factors
d0 &lt;- calcNormFactors(d0)
d0</code></pre>
<pre><code>An object of class &quot;DGEList&quot;
$counts
                   CAF1_V1 CAF1_V2 CAF1_MMC1 CAF1_MMC2 CAF2_V1 CAF2_V2
ENSMUSG00000026193  313278  368553    272989    230151  153189  148543
ENSMUSG00000040152  372316  426808    304702    260705   65276   63117
ENSMUSG00000001506  216965  255796    160159    143006  126065  128695
ENSMUSG00000029661  173567  197616    131673    118101  148318  145185
ENSMUSG00000037742  109253  122688    121259     99121  123691  131161
                   CAF2_MMC1 CAF2_MMC2
ENSMUSG00000026193    136134    143371
ENSMUSG00000040152     52535     58877
ENSMUSG00000001506    116504    107571
ENSMUSG00000029661    118020    114836
ENSMUSG00000037742    107255    119192
28741 more rows ...

$samples
          group lib.size norm.factors
CAF1_V1       1 12175504    0.8866511
CAF1_V2       1 13818374    0.8720662
CAF1_MMC1     1 12434515    0.9373871
CAF1_MMC2     1 10427362    0.9415915
CAF2_V1       1 10341126    1.0687947
CAF2_V2       1 10380767    1.0713266
CAF2_MMC1     1 10245181    1.1215404
CAF2_MMC2     1 10814496    1.1409987

$genes
                              gene_id gene_name gene_chr gene_start  gene_end
ENSMUSG00000026193 ENSMUSG00000026193       Fn1        1   71624679  71692359
ENSMUSG00000040152 ENSMUSG00000040152     Thbs1        2  117942357 117957614
ENSMUSG00000001506 ENSMUSG00000001506    Col1a1       11   94827050  94843868
ENSMUSG00000029661 ENSMUSG00000029661    Col1a2        6    4504814   4541544
ENSMUSG00000037742 ENSMUSG00000037742    Eef1a1        9   78385731  78396433
                   gene_strand gene_length   gene_biotype
ENSMUSG00000026193           -       11685 protein_coding
ENSMUSG00000040152           +        6073 protein_coding
ENSMUSG00000001506           +        6563 protein_coding
ENSMUSG00000029661           +       12426 protein_coding
ENSMUSG00000037742           -        2493 protein_coding
                                                                                         gene_description
ENSMUSG00000026193                                        fibronectin 1 [Source:MGI Symbol;Acc:MGI:95566]
ENSMUSG00000040152                                     thrombospondin 1 [Source:MGI Symbol;Acc:MGI:98737]
ENSMUSG00000001506                            collagen, type I, alpha 1 [Source:MGI Symbol;Acc:MGI:88467]
ENSMUSG00000029661                            collagen, type I, alpha 2 [Source:MGI Symbol;Acc:MGI:88468]
ENSMUSG00000037742 eukaryotic translation elongation factor 1 alpha 1 [Source:MGI Symbol;Acc:MGI:1096881]
28741 more rows ...</code></pre>
<pre class="r"><code># Combining expression and metadata
metadata_expression &lt;- rpkm(d0, gene.length = &quot;gene_length&quot;) %&gt;% t() %&gt;% as.data.frame() %&gt;%
  rownames_to_column(var = &quot;Sample_Treatment&quot;)

metadata_expression$Sample &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 1)
metadata_expression$Treatment &lt;- sapply(strsplit(metadata_expression$Sample_Treatment, &quot;_&quot;), `[`, 2)

long_df &lt;- metadata_expression %&gt;%
  pivot_longer(cols = grep(&quot;ENS&quot;, colnames(metadata_expression), value = T),
               names_to = &quot;ensembl&quot;, values_to = &quot;exp&quot;)

# Adding gene names
long_df$gene &lt;- mapvalues(x = long_df$ensembl,
                          from = gene_info$gene_id,
                          to = gene_info$gene_name)

long_df$Hh_gene &lt;- mapvalues(x = long_df$gene,
                             from = mh_data$Symbol.x,
                             to = mh_data$Symbol.y)

long_df$Treatment_simple &lt;- ifelse(long_df$Treatment %in% c(&quot;V1&quot;, &quot;V2&quot;), &quot;V&quot;, &quot;MMC&quot;)

# Plotting
long_df %&gt;% filter(Hh_gene %in% icafs) %&gt;%
  ggplot(aes(x = Treatment_simple, y = exp, color = Treatment_simple, label = Sample)) +
  geom_violin(alpha = 0.5) +
  geom_point() +
  geom_text(hjust=0, vjust=0) +
  #geom_line(aes(group = Sample), color = &quot;gray20&quot;) +
  facet_wrap(~Hh_gene, scales = &quot;free&quot;) +
  ylab(&quot;iCAF expression (FPKM)&quot;) +
  xlab(&quot;&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-18-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>long_df %&gt;% filter(Hh_gene %in% mycafs) %&gt;%
  ggplot(aes(x = Treatment_simple, y = exp, color = Treatment_simple, label = Sample)) +
  geom_violin(alpha = 0.5) +
  geom_point() +
  geom_text(hjust=0, vjust=0) +
  #geom_line(aes(group = Sample), color = &quot;gray20&quot;) +
  facet_wrap(~Hh_gene, scales = &quot;free&quot;) +
  ylab(&quot;myCAF expression (FPKM)&quot;) +
  xlab(&quot;&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-18-2.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>long_df %&gt;% filter(Hh_gene %in% abcafs) %&gt;%
  ggplot(aes(x = Treatment_simple, y = exp, color = Treatment_simple, label = Sample)) +
  geom_violin(alpha = 0.5) +
  geom_point() +
  geom_text(hjust=0, vjust=0) +
  #geom_line(aes(group = Sample), color = &quot;gray20&quot;) +
  facet_wrap(~Hh_gene, scales = &quot;free&quot;) +
  ylab(&quot;abCAF expression (FPKM)&quot;) +
  xlab(&quot;&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-18-3.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>long_df %&gt;% filter(Hh_gene %in% ficafs) %&gt;%
  ggplot(aes(x = Treatment_simple, y = exp, color = Treatment_simple, label = Sample)) +
  geom_violin(alpha = 0.5) +
  geom_point() +
  geom_text(hjust=0, vjust=0) +
  #geom_line(aes(group = Sample), color = &quot;gray20&quot;) +
  facet_wrap(~Hh_gene, scales = &quot;free&quot;) +
  ylab(&quot;fiCAF expression (FPKM)&quot;) +
  xlab(&quot;&quot;) +
  theme_classic() +
  theme(legend.position = &quot;none&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-18-4.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="pathway-enrichment" class="section level2">
<h2>Pathway enrichment</h2>
<pre class="r"><code># Ordering genes
gse_input &lt;- top.table %&gt;%
  filter(adj.P.Val &lt; 0.1)

gse &lt;- enrichGO(gse_input$gene_id, 
                ont =&quot;BP&quot;, 
                keyType = &quot;ENSEMBL&quot;, 
                minGSSize = 3, 
                maxGSSize = 800, 
                pvalueCutoff = 0.05, 
                universe = top.table$gene_id,
                OrgDb = org.Hs.eg.db, 
                pAdjustMethod = &quot;BH&quot;)

dotplot(gse, showCategory=30, font.size=10, color=&quot;qvalue&quot;) +
  theme(legend.text=element_text(size=5),
        legend.position = &quot;top&quot;,
        legend.box = &quot;vertical&quot;,
        text = element_text(size=6))</code></pre>
<p><img src="figure/CAF_bulkRNAseq.Rmd/unnamed-chunk-19-1.png" width="576" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ComplexHeatmap_2.18.0      viridis_0.6.3             
 [3] viridisLite_0.4.2          Seurat_5.0.1              
 [5] SeuratObject_5.0.2         sp_1.6-1                  
 [7] circlize_0.4.15            RColorBrewer_1.1-3        
 [9] org.Hs.eg.db_3.18.0        AnnotationDbi_1.64.1      
[11] IRanges_2.36.0             S4Vectors_0.40.2          
[13] Biobase_2.62.0             BiocGenerics_0.48.1       
[15] clusterProfiler_4.13.0.004 googlesheets4_1.1.0       
[17] survminer_0.4.9            ggpubr_0.6.0              
[19] survival_3.5-5             contsurvplot_0.2.0        
[21] patchwork_1.1.2            EnhancedVolcano_1.20.0    
[23] ggrepel_0.9.3              plyr_1.8.8                
[25] edgeR_4.0.16               limma_3.58.1              
[27] lubridate_1.9.2            forcats_1.0.0             
[29] stringr_1.5.0              dplyr_1.1.2               
[31] purrr_1.0.2                readr_2.1.4               
[33] tidyr_1.3.0                tibble_3.2.1              
[35] ggplot2_3.4.2              tidyverse_2.0.0           
[37] workflowr_1.7.1           

loaded via a namespace (and not attached):
  [1] spatstat.sparse_3.0-1   matrixStats_1.0.0       fs_1.6.2               
  [4] bitops_1.0-7            enrichplot_1.22.0       doParallel_1.0.17      
  [7] HDO.db_0.99.1           httr_1.4.6              sctransform_0.4.1      
 [10] tools_4.3.0             backports_1.4.1         utf8_1.2.3             
 [13] R6_2.5.1                uwot_0.1.14             lazyeval_0.2.2         
 [16] GetoptLong_1.0.5        withr_2.5.0             gridExtra_2.3          
 [19] progressr_0.13.0        cli_3.6.1               spatstat.explore_3.2-1 
 [22] fastDummies_1.7.3       scatterpie_0.2.1        labeling_0.4.2         
 [25] sass_0.4.6              spatstat.data_3.0-1     survMisc_0.5.6         
 [28] pbapply_1.7-0           ggridges_0.5.4          yulab.utils_0.2.1.002  
 [31] gson_0.1.0              DOSE_3.28.2             parallelly_1.36.0      
 [34] rstudioapi_0.14         RSQLite_2.3.1           generics_0.1.3         
 [37] gridGraphics_0.5-1      shape_1.4.6             spatstat.random_3.1-5  
 [40] ica_1.0-3               car_3.1-2               GO.db_3.18.0           
 [43] Matrix_1.6-5            fansi_1.0.4             abind_1.4-5            
 [46] lifecycle_1.0.3         whisker_0.4.1           yaml_2.3.7             
 [49] carData_3.0-5           qvalue_2.36.0           Rtsne_0.16             
 [52] blob_1.2.4              promises_1.2.0.1        crayon_1.5.2           
 [55] miniUI_0.1.1.1          lattice_0.21-8          cowplot_1.1.1          
 [58] KEGGREST_1.42.0         pillar_1.9.0            knitr_1.43             
 [61] fgsea_1.28.0            rjson_0.2.21            future.apply_1.11.0    
 [64] codetools_0.2-19        leiden_0.4.3            fastmatch_1.1-3        
 [67] glue_1.6.2              getPass_0.2-4           ggfun_0.2.0            
 [70] data.table_1.14.8       vctrs_0.6.5             png_0.1-8              
 [73] treeio_1.26.0           spam_2.9-1              cellranger_1.1.0       
 [76] gtable_0.3.3            cachem_1.0.8            xfun_0.39              
 [79] mime_0.12               tidygraph_1.2.3         gargle_1.4.0           
 [82] iterators_1.0.14        KMsurv_0.1-5            statmod_1.5.0          
 [85] ellipsis_0.3.2          fitdistrplus_1.1-11     ROCR_1.0-11            
 [88] nlme_3.1-162            ggtree_3.10.1           bit64_4.6.0-1          
 [91] RcppAnnoy_0.0.20        GenomeInfoDb_1.38.8     rprojroot_2.0.3        
 [94] bslib_0.5.0             irlba_2.3.5.1           KernSmooth_2.23-21     
 [97] colorspace_2.1-0        DBI_1.1.3               tidyselect_1.2.0       
[100] processx_3.8.1          bit_4.0.5               compiler_4.3.0         
[103] curl_5.0.1              git2r_0.32.0            plotly_4.10.2          
[106] shadowtext_0.1.2        scales_1.2.1            lmtest_0.9-40          
[109] callr_3.7.3             rappdirs_0.3.3          goftest_1.2-3          
[112] digest_0.6.31           spatstat.utils_3.0-3    rmarkdown_2.22         
[115] XVector_0.42.0          htmltools_0.5.5         pkgconfig_2.0.3        
[118] highr_0.10              fastmap_1.1.1           htmlwidgets_1.6.2      
[121] rlang_1.1.6             GlobalOptions_0.1.2     shiny_1.7.4            
[124] farver_2.1.1            jquerylib_0.1.4         zoo_1.8-12             
[127] jsonlite_1.8.5          BiocParallel_1.36.0     GOSemSim_2.28.1        
[130] RCurl_1.98-1.12         magrittr_2.0.3          GenomeInfoDbData_1.2.11
[133] ggplotify_0.1.0         dotCall64_1.0-2         munsell_0.5.0          
[136] Rcpp_1.0.10             reticulate_1.30         ape_5.7-1              
[139] stringi_1.7.12          ggraph_2.1.0            zlibbioc_1.48.2        
[142] MASS_7.3-60             parallel_4.3.0          listenv_0.9.0          
[145] deldir_1.0-9            Biostrings_2.70.3       graphlayouts_1.0.0     
[148] splines_4.3.0           tensor_1.5              hms_1.1.3              
[151] locfit_1.5-9.8          ps_1.7.5                igraph_1.4.3           
[154] spatstat.geom_3.2-1     ggsignif_0.6.4          RcppHNSW_0.5.0         
[157] reshape2_1.4.4          evaluate_0.21           tzdb_0.4.0             
[160] foreach_1.5.2           tweenr_2.0.2            httpuv_1.6.11          
[163] RANN_2.6.1              polyclip_1.10-4         clue_0.3-64            
[166] scattermore_1.2         future_1.32.0           km.ci_0.5-6            
[169] ggforce_0.4.1           broom_1.0.5             xtable_1.8-4           
[172] RSpectra_0.16-1         tidytree_0.4.6          rstatix_0.7.2          
[175] later_1.3.1             googledrive_2.1.0       aplot_0.2.9            
[178] memoise_2.0.1           cluster_2.1.4           timechange_0.2.0       
[181] globals_0.16.2         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
