<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="heinin" />

<meta name="date" content="2024-09-17" />

<title>Xenium data processing</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIPAC_spatial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Xenium data processing</h1>
<h4 class="author">heinin</h4>
<h4 class="date">2024-09-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>PIPAC_spatial/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240917code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240917)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240917code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240917)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/
</td>
<td style="text-align:left;">
.
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PIPAC_spatial/code/colors_themes.R
</td>
<td style="text-align:left;">
code/colors_themes.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong668678e">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 668678e
</a>
</p>
</div>
<div id="strongRepositoryversionstrong668678e"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 668678e.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  analysis/Xenium_processing.Rmd
    Untracked:  analysis/annotate.Rmd
    Untracked:  analysis/annotation.Rmd
    Untracked:  analysis/post_clustering.Rmd
    Untracked:  analysis/splitting_samples.Rmd
    Untracked:  code/PIPAC_colors_themes.R
    Untracked:  code/anndata_to_seurat.R
    Untracked:  code/plot_functions.R
    Untracked:  code/rapids_pipeline/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="packages-and-environment-variables" class="section level3">
<h3>Packages and environment variables</h3>
<pre class="r"><code>suppressPackageStartupMessages({
  library(workflowr)
  library(arrow)
  library(Seurat)
  library(SeuratObject)
  library(SeuratDisk)
  library(tidyverse)
  library(tibble)
  library(ggplot2)
  library(ggpubr)
  library(ggrepel)
  library(workflowr)})</code></pre>
</div>
<div id="environment-variables-and-helper-functions"
class="section level3">
<h3>Environment variables and helper functions</h3>
<pre class="r"><code>setwd(&quot;/home/hnatri/PIPAC_spatial/&quot;)
set.seed(9999)
options(scipen = 99999)
options(ggrepel.max.overlaps = Inf)

# Colors, themes, and cell type markers
#source(&quot;/home/hnatri/PIPAC_spatial/code/colors_themes.R&quot;)

# A simplified VlnPlot
BetterVlnPlot &lt;- function(data, features, group_by, ylim = NA){
  VlnPlot(data, pt.size = 0, features = features, 
          group.by = group_by, y.max = ylim) + labs(x = &quot;&quot;) + NoLegend()
}

# DimPlots of before and after for each sample
DimPlotCompare &lt;- function(tma){
  bf_cells &lt;- ncol(subset(merged_spatial_unfiltered, subset = TMA == tma))
  a &lt;- DimPlot(subset(merged_spatial_unfiltered, subset = TMA == tma),
               reduction = &quot;sp&quot;) + NoLegend() +
    labs(title = paste0(tma, &quot;, Unfiltered, &quot;, bf_cells, &quot; nuclei&quot;))
  
  aft_cells &lt;- ncol(subset(merged_spatial, subset = TMA == tma))
  b &lt;- DimPlot(subset(merged_spatial, subset = TMA == tma),
               reduction = &quot;sp&quot;) + NoLegend() +
    labs(title = paste0(tma, &quot;, Filtered, &quot;, aft_cells, &quot; nuclei&quot;))
  ggarrange(a,b)
}</code></pre>
</div>
<div id="import-data" class="section level3">
<h3>Import data</h3>
<pre class="r"><code># /tgen_labs/banovich/xenium_run_folders/output-XETG00048__0034123__MR_PIPACTMA1__20240911_210055
# /tgen_labs/banovich/xenium_run_folders/PIPACTMA
data_dir &lt;- &quot;/tgen_labs/banovich/xenium_run_folders/PIPACTMA/&quot;
tma_list &lt;- c(
  TMA1 = &quot;output-XETG00048__0034123__MR_PIPACTMA1__20240911__210055&quot;,
  TMA2 = &quot;output-XETG00048__0034221__MR_PIPACTMA2-Rerun__20241107__213721&quot;,
  TMA3 = &quot;output-XETG00048__0024823__MR_PIPACTMA3-Rerun__20241115__191438&quot;,
  TMA4 = &quot;output-XETG00048__0033982__MR_PIPACTMA4__20240920__231140&quot;,
  TMA5 = &quot;output-XETG00048__0033981__MR_PIPACTMA5__20240920__231140&quot;,
  TMA6 = &quot;output-XETG00048__0034149__MR_PIPACTMA6__20240923__231239&quot;,
  TMA7 = &quot;output-XETG00048__0034152__MR_PIPACTMA7__20240923__231239&quot;,
  TMA8 = &quot;output-XETG00048__0033448__MR_PIPACTMA8__20241224__204450&quot;,
  TMA9 = &quot;output-XETG00048__0041382__MR_PIPACTMA9__20250109__205856&quot;)

# Get subdirectory names for obtaining file paths
subdirs &lt;- unname(tma_list)

# Get transcript counts and metadata
all_files &lt;- list.files(file.path(data_dir, subdirs), full.names = TRUE)
h5_files &lt;- all_files[grep(&quot;.h5&quot;, all_files)]
transcript_files &lt;- all_files[grep(&quot;transcripts.parquet&quot;, all_files)]
meta_files &lt;- all_files[grep(&quot;cells.csv.gz&quot;, all_files)]

# Get sample IDs
tma_ids &lt;- names(tma_list)

# Read in files
counts &lt;- lapply(h5_files, Read10X_h5)

transcripts &lt;- lapply(transcript_files, function(XX) {
  read_parquet(XX) })

metadata &lt;- lapply(meta_files, function(XX) {
  tmp_meta &lt;- read.delim(XX, sep = &quot;,&quot;, colClasses = c(cell_id = &quot;character&quot;))
  rownames(tmp_meta) &lt;- tmp_meta$cell_id
  tmp_meta })

# Rename files in lists
tma_ids &lt;- unlist(lapply(str_split(meta_files, &quot;__&quot;), function(XX) { XX[3] }))
names(counts) &lt;- tma_ids
names(transcripts) &lt;- tma_ids
names(metadata) &lt;- tma_ids</code></pre>
</div>
<div
id="get-transcripts-that-only-overlap-the-nucleus-create-cell-x-gene-matrix-and-count-the-number-of-blanks-per-cell"
class="section level3">
<h3>Get transcripts that only overlap the nucleus, create cell x gene
matrix, and count the number of blanks per cell</h3>
<pre class="r"><code>all_transcripts &lt;- list()
nuc_transcripts &lt;- list()
updated_metadata &lt;- list()
for (tma in tma_ids) {
  message(paste(&quot;Getting nuclei counts for sample&quot;, tma))
  
  # Filter out low quality transcripts 
  all_transcripts[[tma]] &lt;- transcripts[[tma]][transcripts[[tma]]$qv &gt; 20, ]
  
  # Find transcripts that overlap a nucleus
  nuc_transcripts[[tma]] &lt;- transcripts[[tma]][transcripts[[tma]]$overlaps_nucleus == &quot;1&quot;, ]
  
  # Create cell x gene dataframe
  nuc_transcripts[[tma]] &lt;- as.data.frame(table(nuc_transcripts[[tma]]$cell_id, 
                                               nuc_transcripts[[tma]]$feature_name))
  names(nuc_transcripts[[tma]]) &lt;- c(&quot;cell_id&quot;, &quot;feature_name&quot;, &quot;Count&quot;)
  nuc_transcripts[[tma]] &lt;- nuc_transcripts[[tma]] %&gt;% 
    pivot_wider(names_from = &quot;feature_name&quot;, values_from = &quot;Count&quot;)
  
  # Get blanks count per nucleus
  blank_nuc_ids &lt;- nuc_transcripts[[tma]]$cell_id
  blank_nuc_mat &lt;- nuc_transcripts[[tma]][, grep(&quot;BLANK&quot;, 
                                                colnames(nuc_transcripts[[tma]]))]
  blank_nuc_counts &lt;- as.data.frame(rowSums(blank_nuc_mat))
  blank_nuc_counts$cell_id &lt;- blank_nuc_ids
  
  # Remove negative controls and convert to cell x gene matrix
  nuc_transcripts[[tma]] &lt;- nuc_transcripts[[tma]][, grep(&quot;NegControl&quot;, 
                                                        colnames(nuc_transcripts[[tma]]), 
                                                        invert = TRUE)]
  nuc_transcripts[[tma]] &lt;- nuc_transcripts[[tma]][, grep(&quot;BLANK&quot;, 
                                                        colnames(nuc_transcripts[[tma]]), 
                                                        invert = TRUE)]
  keep_cells &lt;- nuc_transcripts[[tma]]$cell_id
  nuc_transcripts[[tma]] &lt;- as.data.frame(nuc_transcripts[[tma]])
  rownames(nuc_transcripts[[tma]]) &lt;- keep_cells
  nuc_transcripts[[tma]] &lt;- nuc_transcripts[[tma]][, -1]
  nuc_transcripts[[tma]] &lt;- as.matrix(t(nuc_transcripts[[tma]]))
  
  # Subset nuclear metadata to &quot;cells&quot; with transcripts that overlap nuclei
  updated_metadata[[tma]] &lt;- metadata[[tma]][metadata[[tma]]$cell_id %in% keep_cells, ]
  
  # Add blank counts to metadata
  updated_metadata[[tma]] &lt;- full_join(updated_metadata[[tma]], blank_nuc_counts,
                                      by = &quot;cell_id&quot;)
  updated_metadata[[tma]] &lt;- updated_metadata[[tma]] %&gt;%
    dplyr::rename(num.blank = `rowSums(blank_nuc_mat)`)
  rownames(updated_metadata[[tma]]) &lt;- updated_metadata[[tma]]$cell_id
}</code></pre>
</div>
<div id="create-seurat-objects" class="section level3">
<h3>Create Seurat objects</h3>
<pre class="r"><code>obj_list &lt;- list()
obj_list &lt;- sapply(tma_ids, function(XX) {
  # Create a Seurat object containing the RNA adata
  sobj &lt;- CreateSeuratObject(counts = nuc_transcripts[[XX]], 
                             assay = &quot;RNA&quot;)
  
  # Add metadata
  sobj &lt;- AddMetaData(sobj, metadata = updated_metadata[[XX]])
  sobj$TMA &lt;- XX
  #sobj$tma &lt;- tmas[[XX]]
  #sobj$run &lt;- run_ids[[XX]]
  
  # Calculate percent blank
  sobj$percent.blank &lt;- sobj$num.blank/(sobj$nCount_RNA + sobj$num.blank)*100
  
  # Remove cells with 0 nCount_RNA
  sobj &lt;- subset(sobj, subset = nCount_RNA != 0)
  
  # Rename cells to add TMA name as prefix
  cellid &lt;- gsub(&quot;MR_PIPAC&quot;, &quot;&quot;, XX)
  cellid &lt;- gsub(&quot;-Rerun&quot;, &quot;&quot;, cellid)
  sobj &lt;- RenameCells(sobj, add.cell.id = cellid)
  
  # Add spatial coordinates as dimension reduction objects
  #position_xy &lt;- cbind(sobj$adj_x_centroid, sobj$adj_y_centroid)
  position_xy &lt;- cbind(sobj$x_centroid, sobj$y_centroid)
  row.names(position_xy) &lt;- row.names(sobj@meta.data)
  colnames(position_xy) &lt;- c(&quot;SP_1&quot;, &quot;SP_2&quot;)
  sobj[[&quot;sp&quot;]] &lt;- CreateDimReducObject(embeddings = position_xy, key = &quot;SP_&quot;,
                                       assay = DefaultAssay(sobj))
  obj_list[[XX]] &lt;- sobj

})

#saveRDS(obj_list, &quot;/scratch/hnatri/PIPAC/obj_list.rds&quot;)

# Get TMA IDs
tma_ids &lt;- names(obj_list)</code></pre>
</div>
<div id="visualize" class="section level3">
<h3>Visualize</h3>
<pre class="r"><code>obj_list &lt;- readRDS(&quot;/scratch/hnatri/PIPAC/obj_list.rds&quot;)

# Merge objects (cannot do spatial DimPlots for this)
#merged_spatial_unfiltered &lt;- obj_list[[1]]
merged_spatial_unfiltered &lt;- merge(x = obj_list[[1]], y = obj_list[2:length(obj_list)])

# Add spatial dimension reduction object separately
position_xy &lt;- cbind(merged_spatial_unfiltered$x_centroid,
                     merged_spatial_unfiltered$y_centroid)
row.names(position_xy) &lt;- row.names(merged_spatial_unfiltered@meta.data)
colnames(position_xy) &lt;- c(&quot;SP_1&quot;, &quot;SP_2&quot;)
merged_spatial_unfiltered[[&quot;sp&quot;]] &lt;- CreateDimReducObject(
  embeddings = position_xy, key = &quot;SP_&quot;, assay = DefaultAssay(merged_spatial_unfiltered))

DimPlot(merged_spatial_unfiltered,
        reduction = &quot;sp&quot;,
        group.by = &quot;TMA&quot;,
        label = F,
        split.by = &quot;TMA&quot;,
        ncol = 3) +
  coord_fixed() + NoLegend()</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-6-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#saveRDS(merged_spatial_unfiltered, &quot;/scratch/hnatri/PIPAC/merged_spatial_unfiltered.rds&quot;)</code></pre>
</div>
<div id="add-cell-level-count-data" class="section level3">
<h3>Add cell level count data</h3>
<pre class="r"><code># Get sample IDs
tma_ids &lt;- names(obj_list)

cell_obj_list &lt;- list()
cell_obj_list &lt;- sapply(tma_ids, function(XX) {
  message(paste(&quot;Creating cell Seurat object for TMA&quot;, XX))
  
  # Create a Seurat object containing the RNA cell information
  sobj &lt;- CreateSeuratObject(counts = counts[[XX]]$`Gene Expression`,
                             assay = &quot;RNA&quot;)
  rownames(metadata[[XX]]) &lt;- metadata[[XX]]$cell_id
  sobj &lt;- AddMetaData(sobj, metadata = metadata[[XX]])
  
  # Rename cells to add sample ID as prefix
  cellid &lt;- gsub(&quot;MR_PIPAC&quot;, &quot;&quot;, XX)
  cellid &lt;- gsub(&quot;-Rerun&quot;, &quot;&quot;, cellid)
  sobj &lt;- RenameCells(sobj, add.cell.id = cellid)

  cell_obj_list[[XX]] &lt;- sobj
})

# Merge cell information
#cell_merged &lt;- cell_obj_list[[1]]
cell_merged &lt;- merge(cell_obj_list[[1]], y = cell_obj_list[2:length(cell_obj_list)])

# Add cell information to nuclei object
cell_merged &lt;- JoinLayers(cell_merged)
cell_count_matrix &lt;- cell_merged@assays$RNA$counts
keep_cells &lt;- colnames(merged_spatial_unfiltered)
cell_count_matrix &lt;- cell_count_matrix[, keep_cells]
merged_spatial_unfiltered[[&quot;cell_RNA&quot;]] &lt;- CreateAssayObject(counts = cell_count_matrix)

#saveRDS(merged_spatial_unfiltered, &quot;/scratch/hnatri/PIPAC/merged_spatial_unfiltered.rds&quot;)</code></pre>
</div>
<div id="qc" class="section level3">
<h3>QC</h3>
<pre class="r"><code>merged_spatial_unfiltered &lt;- readRDS(&quot;/scratch/hnatri/PIPAC/merged_spatial_unfiltered.rds&quot;)

# Number of cells per TMA before filtering
summary(as.factor(merged_spatial_unfiltered$TMA))</code></pre>
<pre><code>      MR_PIPACTMA1 MR_PIPACTMA2-Rerun MR_PIPACTMA3-Rerun       MR_PIPACTMA4 
            163174             195030             239629             170103 
      MR_PIPACTMA5       MR_PIPACTMA6       MR_PIPACTMA7       MR_PIPACTMA8 
            178743             186174             255775             136737 
      MR_PIPACTMA9 
            137688 </code></pre>
<pre class="r"><code>merged_spatial_unfiltered@meta.data %&gt;%
  ggplot(aes(x = TMA)) +
  geom_bar(stat = &quot;count&quot;) +
  theme_minimal() +
  RotatedAxis()</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Percent.blank
merged_spatial_unfiltered@meta.data %&gt;%
  ggplot(aes(x = percent.blank, fill = TMA)) +
  geom_histogram(bins = 50, show.legend = FALSE, color = &quot;black&quot;) +
  theme_classic() +
  theme(title = element_text(color = &quot;black&quot;), 
        axis.text = element_text(color = &quot;black&quot;)) +
  facet_wrap(~TMA, scales = &quot;free&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-9-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nCount_RNA
merged_spatial_unfiltered@meta.data %&gt;%
  ggplot(aes(x = nCount_RNA, fill = TMA)) +
  geom_histogram(bins = 50, show.legend = FALSE, color = &quot;black&quot;) +
  theme_classic() +
  theme(title = element_text(color = &quot;black&quot;), 
        axis.text = element_text(color = &quot;black&quot;)) +
  facet_wrap(~TMA, scales = &quot;free&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-9-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nucleus_area
merged_spatial_unfiltered@meta.data %&gt;%
  ggplot(aes(x = nucleus_area, fill = TMA)) +
  geom_histogram(bins = 50, show.legend = FALSE, color = &quot;black&quot;) +
  theme_classic() +
  theme(title = element_text(color = &quot;black&quot;), 
        axis.text = element_text(color = &quot;black&quot;)) +
  facet_wrap(~TMA, scales = &quot;free&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-9-3.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>merged_spatial_unfiltered$TMA &lt;- factor(merged_spatial_unfiltered$TMA,
                                           levels = rev(sort(unique(merged_spatial_unfiltered$TMA))))</code></pre>
<pre class="r"><code>BetterVlnPlot(merged_spatial_unfiltered,
              features = &quot;percent.blank&quot;, group_by = &quot;TMA&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>BetterVlnPlot(merged_spatial_unfiltered,
              features = &quot;nCount_RNA&quot;, group_by = &quot;TMA&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-10-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>BetterVlnPlot(merged_spatial_unfiltered,
              features = &quot;nFeature_RNA&quot;, group_by = &quot;TMA&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-10-3.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>BetterVlnPlot(merged_spatial_unfiltered,
              features = &quot;nucleus_area&quot;, group_by = &quot;TMA&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-10-4.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## nCount_RNA vs. percent.blank
#smoothScatter(merged_spatial_unfiltered@meta.data$percent.blank,
#              log(merged_spatial_unfiltered@meta.data$nCount_RNA),
#              cex = 0.5, pch = 16)
#abline(v = 4, h = log(12), lty = &quot;dashed&quot;, col = &quot;black&quot;)
#text(5, 5, col = &quot;black&quot;, adj = c(0, -.1),
#     &quot;nCount_RNA &gt;= 12 &amp; percent.blank &lt;= 4&quot;)
#
## nFeature_RNA vs. percent.blank
#smoothScatter(merged_spatial_unfiltered@meta.data$percent.blank,
#              log(merged_spatial_unfiltered@meta.data$nFeature_RNA),
#              cex = 0.5, pch = 16)
#abline(v = 4, h = log(10), lty = &quot;dashed&quot;, col = &quot;black&quot;)
#text(5, 4, col = &quot;black&quot;, adj = c(0, -.1),
#     &quot;nFeature_RNA &gt;= 10 &amp; percent.blank &lt;= 4&quot;)

# nCount_RNA vs. nFeature_RNA
smoothScatter(log(merged_spatial_unfiltered$nCount_RNA),
              log(merged_spatial_unfiltered$nFeature_RNA),
              cex = 0.5, pch = 16)
abline(v = log(10), h = log(10), lty = &quot;dashed&quot;, col = &quot;black&quot;)
text(0.3, 4.6, col = &quot;black&quot;, adj = c(0, -.1),
     &quot;nCount_RNA &gt;= 10 &amp; nFeature_RNA &gt;= 10&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-11-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>smoothScatter(merged_spatial_unfiltered$nCount_RNA,
              merged_spatial_unfiltered$nFeature_RNA,
              cex = 0.5, pch = 16)
abline(v = log(10), h = log(10), lty = &quot;dashed&quot;, col = &quot;black&quot;)
text(0.3, 4.6, col = &quot;black&quot;, adj = c(0, -.1),
     &quot;nCount_RNA &gt;= 10 &amp; nFeature_RNA &gt;= 10&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-11-2.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nCount RNA vs. nucleus_area
smoothScatter(merged_spatial_unfiltered$nucleus_area,
              log(merged_spatial_unfiltered$nCount_RNA),
              cex = 0.5, pch = 16)
abline(v = c(5, 80), h = log(10), lty = &quot;dashed&quot;, col = &quot;black&quot;)
text(120, 0.7, col = &quot;black&quot;, adj = c(0, -.1),
     &quot;nCount_RNA &gt;= 10 &amp; nucleus_area between 5-80&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-11-3.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code># nFeature RNA vs. nucleus_area
smoothScatter(merged_spatial_unfiltered$nucleus_area,
              log(merged_spatial_unfiltered$nFeature_RNA),
              cex = 0.5, pch = 16)
abline(v = c(5, 80), h = log(10), lty = &quot;dashed&quot;, col = &quot;black&quot;)
text(120, 0.4, col = &quot;black&quot;, adj = c(0, -.1),
     &quot;nFeature_RNA &gt;= 10 &amp; &amp; nucleus_area between 5-80&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-11-4.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>min(merged_spatial_unfiltered$nucleus_area)</code></pre>
<pre><code>[1] 1.399844</code></pre>
<pre class="r"><code>max(merged_spatial_unfiltered$nucleus_area)</code></pre>
<pre><code>[1] 308.4624</code></pre>
</div>
<div id="filter" class="section level3">
<h3>Filter</h3>
<pre class="r"><code># Filter merged and individual data
merged_spatial &lt;- subset(merged_spatial_unfiltered,
                          subset = nCount_RNA &gt;= 10 &amp; nFeature_RNA &gt;= 10 &amp;
                            percent.blank &lt;= 5 &amp; 
                            nucleus_area &gt;= 5 &amp; nucleus_area &lt;= 80)

# Number of nuclei before and after filtering
bf_cells &lt;- table(merged_spatial_unfiltered$TMA)
aft_cells &lt;- table(merged_spatial$TMA)
diff_cells &lt;- bf_cells - aft_cells
prop_kept_cells &lt;- round(aft_cells/bf_cells*100, 2)
prop_kept_cells</code></pre>
<pre><code>
      MR_PIPACTMA9       MR_PIPACTMA8       MR_PIPACTMA7       MR_PIPACTMA6 
             62.12              57.79              66.89              70.08 
      MR_PIPACTMA5       MR_PIPACTMA4 MR_PIPACTMA3-Rerun MR_PIPACTMA2-Rerun 
             52.42              60.45              69.48              68.95 
      MR_PIPACTMA1 
             61.16 </code></pre>
<pre class="r"><code># DimPlots before and after filtering
DimPlotCompare(&quot;MR_PIPACTMA1&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-12-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlotCompare(&quot;MR_PIPACTMA2-Rerun&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-12-2.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlotCompare(&quot;MR_PIPACTMA3-Rerun&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-12-3.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlotCompare(&quot;MR_PIPACTMA4&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-12-4.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlotCompare(&quot;MR_PIPACTMA5&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-12-5.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlotCompare(&quot;MR_PIPACTMA6&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-12-6.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlotCompare(&quot;MR_PIPACTMA7&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-12-7.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlotCompare(&quot;MR_PIPACTMA8&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-12-8.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlotCompare(&quot;MR_PIPACTMA9&quot;)</code></pre>
<p><img src="figure/Xenium_processing.Rmd/unnamed-chunk-12-9.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Saving the object for clustering with ScanPy
merged_spatial &lt;- JoinLayers(merged_spatial)
saveRDS(merged_spatial, &quot;/scratch/hnatri/PIPAC/merged_spatial_filtered.rds&quot;)

# To build on command line, run Rscript -e &quot;rmarkdown::render(&#39;/home/hnatri/PIPAC_Xenium/analysis/Xenium_processing.Rmd&#39;)&quot;
# Then &quot;mv *.html /home/hnatri/PIPAC_Xenium/docs/&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggrepel_0.9.3         ggpubr_0.6.0          lubridate_1.9.2      
 [4] forcats_1.0.0         stringr_1.5.0         dplyr_1.1.2          
 [7] purrr_1.0.2           readr_2.1.4           tidyr_1.3.0          
[10] tibble_3.2.1          ggplot2_3.4.2         tidyverse_2.0.0      
[13] SeuratDisk_0.0.0.9021 Seurat_5.0.1          SeuratObject_5.0.2   
[16] sp_1.6-1              arrow_12.0.0          workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3     rstudioapi_0.14        jsonlite_1.8.5        
  [4] magrittr_2.0.3         ggbeeswarm_0.7.2       spatstat.utils_3.0-3  
  [7] farver_2.1.1           rmarkdown_2.22         fs_1.6.2              
 [10] vctrs_0.6.5            ROCR_1.0-11            spatstat.explore_3.2-1
 [13] rstatix_0.7.2          htmltools_0.5.5        broom_1.0.4           
 [16] sass_0.4.6             sctransform_0.4.1      parallelly_1.36.0     
 [19] KernSmooth_2.23-21     bslib_0.4.2            htmlwidgets_1.6.2     
 [22] ica_1.0-3              plyr_1.8.8             plotly_4.10.2         
 [25] zoo_1.8-12             cachem_1.0.8           whisker_0.4.1         
 [28] igraph_1.4.3           mime_0.12              lifecycle_1.0.3       
 [31] pkgconfig_2.0.3        Matrix_1.6-5           R6_2.5.1              
 [34] fastmap_1.1.1          fitdistrplus_1.1-11    future_1.32.0         
 [37] shiny_1.7.4            digest_0.6.31          colorspace_2.1-0      
 [40] patchwork_1.1.2        ps_1.7.5               rprojroot_2.0.3       
 [43] tensor_1.5             RSpectra_0.16-1        irlba_2.3.5.1         
 [46] labeling_0.4.2         progressr_0.13.0       timechange_0.2.0      
 [49] fansi_1.0.4            spatstat.sparse_3.0-1  httr_1.4.6            
 [52] polyclip_1.10-4        abind_1.4-5            compiler_4.3.0        
 [55] withr_2.5.0            bit64_4.0.5            backports_1.4.1       
 [58] carData_3.0-5          fastDummies_1.7.3      highr_0.10            
 [61] ggsignif_0.6.4         MASS_7.3-60            tools_4.3.0           
 [64] vipor_0.4.5            lmtest_0.9-40          beeswarm_0.4.0        
 [67] httpuv_1.6.11          future.apply_1.11.0    goftest_1.2-3         
 [70] glue_1.6.2             callr_3.7.3            nlme_3.1-162          
 [73] promises_1.2.0.1       grid_4.3.0             Rtsne_0.16            
 [76] getPass_0.2-4          cluster_2.1.4          reshape2_1.4.4        
 [79] generics_0.1.3         hdf5r_1.3.8            gtable_0.3.3          
 [82] spatstat.data_3.0-1    tzdb_0.4.0             hms_1.1.3             
 [85] data.table_1.14.8      car_3.1-2              utf8_1.2.3            
 [88] spatstat.geom_3.2-1    RcppAnnoy_0.0.20       RANN_2.6.1            
 [91] pillar_1.9.0           spam_2.9-1             RcppHNSW_0.5.0        
 [94] later_1.3.1            splines_4.3.0          lattice_0.21-8        
 [97] survival_3.5-5         bit_4.0.5              deldir_1.0-9          
[100] tidyselect_1.2.0       miniUI_0.1.1.1         pbapply_1.7-0         
[103] knitr_1.43             git2r_0.32.0           gridExtra_2.3         
[106] scattermore_1.2        xfun_0.39              matrixStats_1.0.0     
[109] stringi_1.7.12         lazyeval_0.2.2         yaml_2.3.7            
[112] evaluate_0.21          codetools_0.2-19       cli_3.6.1             
[115] uwot_0.1.14            xtable_1.8-4           reticulate_1.29       
[118] munsell_0.5.0          processx_3.8.1         jquerylib_0.1.4       
[121] Rcpp_1.0.10            globals_0.16.2         spatstat.random_3.1-5 
[124] png_0.1-8              ggrastr_1.0.2          parallel_4.3.0        
[127] ellipsis_0.3.2         assertthat_0.2.1       dotCall64_1.0-2       
[130] listenv_0.9.0          viridisLite_0.4.2      scales_1.2.1          
[133] ggridges_0.5.4         crayon_1.5.2           leiden_0.4.3          
[136] rlang_1.1.1            cowplot_1.1.1         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
